{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package vader_lexicon to\n",
      "[nltk_data]     C:\\Users\\iKhan\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package vader_lexicon is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import nltk\n",
    "nltk.download('vader_lexicon')\n",
    "\n",
    "import gensim\n",
    "import gensim.corpora as corpora\n",
    "from gensim.utils import simple_preprocess\n",
    "from gensim.models import CoherenceModel\n",
    "from sklearn.feature_selection import SelectFromModel\n",
    "from nltk.stem import PorterStemmer\n",
    "\n",
    "\n",
    "from sklearn import model_selection\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.tree import DecisionTreeClassifier\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "from sklearn.discriminant_analysis import LinearDiscriminantAnalysis\n",
    "from sklearn.naive_bayes import GaussianNB\n",
    "from sklearn.svm import SVC\n",
    "from sklearn.ensemble import GradientBoostingClassifier\n",
    "from sklearn.metrics import classification_report, confusion_matrix, roc_curve, auc\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " dataset has 284 samples with 2 features each.\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    df = pd.read_csv(\"Abstracts_TP_TN.csv\")#, sep=\",\")\n",
    "    print(\" dataset has {} samples with {} features each.\".format(*df.shape))\n",
    "except:\n",
    "    print(\"Dataset could not be loaded. Is the dataset missing?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['pubmed_id;abstract;class;', 'Unnamed: 1'], dtype='object')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>Andrographolide, the major labdane diterpenoid...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>DGAT2 plays a critical role in hepatic triglyc...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>Despite the success of imatinib at inhibiting ...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>Two benzazaborinine analogues of propranolol w...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>Our recent study has shown that the natural pr...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          0                                                  1   2 3\n",
       "0  26561364  Andrographolide, the major labdane diterpenoid...  -1  \n",
       "1  26561979  DGAT2 plays a critical role in hepatic triglyc...  -1  \n",
       "2  26562217  Despite the success of imatinib at inhibiting ...  -1  \n",
       "3  26565745  Two benzazaborinine analogues of propranolol w...  -1  \n",
       "4  26572282  Our recent study has shown that the natural pr...  -1  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_tidy = df.rename(columns = {'pubmed_id;abstract;class;': 'col1', 'Unnamed: 1': 'col2'}, inplace = False)\n",
    "#df_tidy.head()\n",
    "df_temp = df_tidy.col1.str.split(\";\",expand=True) \n",
    "df_temp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "RangeIndex(start=0, stop=4, step=1)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_temp.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>target_ML_model</th>\n",
       "      <th>dlt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>Andrographolide, the major labdane diterpenoid...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>DGAT2 plays a critical role in hepatic triglyc...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>Despite the success of imatinib at inhibiting ...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>Two benzazaborinine analogues of propranolol w...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>Our recent study has shown that the natural pr...</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  pubmed_id                                           abstract  \\\n",
       "0  26561364  Andrographolide, the major labdane diterpenoid...   \n",
       "1  26561979  DGAT2 plays a critical role in hepatic triglyc...   \n",
       "2  26562217  Despite the success of imatinib at inhibiting ...   \n",
       "3  26565745  Two benzazaborinine analogues of propranolol w...   \n",
       "4  26572282  Our recent study has shown that the natural pr...   \n",
       "\n",
       "  target_ML_model dlt  \n",
       "0              -1      \n",
       "1              -1      \n",
       "2              -1      \n",
       "3              -1      \n",
       "4              -1      "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ready = df_temp.rename(columns = {0: 'pubmed_id', 1: 'abstract', 2: 'target_ML_model', 3: 'dlt'}, inplace = False)\n",
    "df_ready.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "del df_ready['dlt']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ready = df_ready.drop(df_ready[df_ready['target_ML_model'] == '1'].sample(frac=0.98).index)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>target_ML_model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>Andrographolide, the major labdane diterpenoid...</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>DGAT2 plays a critical role in hepatic triglyc...</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>Despite the success of imatinib at inhibiting ...</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>Two benzazaborinine analogues of propranolol w...</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>Our recent study has shown that the natural pr...</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  pubmed_id                                           abstract target_ML_model\n",
       "0  26561364  Andrographolide, the major labdane diterpenoid...              -1\n",
       "1  26561979  DGAT2 plays a critical role in hepatic triglyc...              -1\n",
       "2  26562217  Despite the success of imatinib at inhibiting ...              -1\n",
       "3  26565745  Two benzazaborinine analogues of propranolol w...              -1\n",
       "4  26572282  Our recent study has shown that the natural pr...              -1"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ready.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0xcef3f98>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAW4AAAD4CAYAAADM6gxlAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjAsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+17YcXAAAKi0lEQVR4nO3df6jd913H8dfbxFUyWbqazXVJ8XZSNuuqruSPVv1D/MHaZVQF/2gZrGCh/wy2iUMTCoL/RRSdQp0WnQUpVaxTS4uOUQf+M+pudGsyu2wdi1viZjammViQrvv4x/lGr+m9JjfL957zbh4PuOSe7/cmnzefnvPsud9zf9QYIwD08W3LHgCA7RFugGaEG6AZ4QZoRrgBmtk99wL79u0ba2trcy8D8LJy7Nixr44xXrPZudnDvba2lvX19bmXAXhZqap/3uqcSyUAzQg3QDPCDdCMcAM0I9wAzQg3QDPCDdCMcAM0I9wAzQg3QDPCDdCMcAM0I9wAzQg3QDPCDdCMcAM0M/svUjh+5lzWDj+56blTRw/NvTzAy45n3ADNCDdAM8IN0IxwAzQj3ADNCDdAM8IN0IxwAzQj3ADNCDdAM8IN0IxwAzQj3ADNCDdAM9sOd1V9sKrOVtWJOQYC4P93Oc+4H05yxxWeA4BLtO1wjzH+LsnXZpgFgEvgGjdAM7OEu6rur6r1qlp/8flzcywBcNWaJdxjjIfGGAfHGAd37dk7xxIAVy2XSgCauZwvB3w0yceSvLGqTlfVfVd+LAC2snu7f2GMcc8cgwBwaVwqAWhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhm2z8dcLtu2b8360cPzb0MwFXDM26AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhm99wLHD9zLmuHn5x7mS2dOnpoaWsDzMEzboBmhBugGeEGaEa4AZoRboBmhBugGeEGaEa4AZoRboBmhBugGeEGaEa4AZoRboBmhBugGeEGaEa4AZoRboBmZgl3Vd1fVetVtf7i8+fmWALgqjVLuMcYD40xDo4xDu7as3eOJQCuWi6VADQj3ADNCDdAM8IN0IxwAzQj3ADNCDdAM8IN0IxwAzQj3ADNCDdAM8IN0IxwAzQj3ADNCDdAM8IN0MzuuRe4Zf/erB89NPcyAFcNz7gBmhFugGaEG6AZ4QZoRrgBmhFugGaEG6AZ4QZoRrgBmhFugGaEG6AZ4QZoRrgBmhFugGaEG6AZ4QZoRrgBmhFugGaEG6AZ4QZoRrgBmhFugGaEG6AZ4QZoRrgBmhFugGaEG6AZ4QZoRrgBmhFugGaEG6AZ4QZoRrgBmhFugGaEG6CZ3XMvcPzMuawdfnLuZQBWyqmjh2b7tz3jBmhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhGuAGaEW6AZoQboBnhBmhGuAGa2Va4q+pNVfWxqvqvqnrfXEMBsLXt/jzuryV5d5KfmWEWAC7Btp5xjzHOjjE+nuSFmeYB4CJc4wZoZpZwV9X9VbVeVesvPn9ujiUArloXDXdVvauqPjG9vf5S/tExxkNjjINjjIO79uz91qcE4H9c9MXJMcaDSR7cgVkAuATb+qqSqnpdkvUkr0ryzap6b5Kbxxhfn2M4AF5qW+EeY3w5yYGZZgHgEviqEoBmhBugGeEGaEa4AZoRboBmhBugGeEGaEa4AZoRboBmhBugGeEGaEa4AZoRboBmtvvLgrftlv17s3700NzLAFw1POMGaEa4AZoRboBmhBugGeEGaEa4AZoRboBmhBugGeEGaEa4AZoRboBmhBugGeEGaEa4AZoRboBmhBugGeEGaKbGGPMuUPUfSU7OusiVty/JV5c9xDaZeWeYeWeYOfmeMcZrNjsx+68uS3JyjHFwB9a5Yqpq3czzM/POMPPO2MmZXSoBaEa4AZrZiXA/tANrXGlm3hlm3hlm3hk7NvPsL04CcGW5VALQjHADNDNruKvqjqo6WVXPVdXhOde6XFV1Q1V9tKqerapPVdV7puPXVdVHquqz05+vXvasG1XVrqr6x6p6Yrp9Y1U9Pc37p1X1imXPuFFVXVtVj1XVp6e9vr3BHv/CdJ84UVWPVtV3rNo+V9UHq+psVZ3YcGzTfa2F35kej89U1a0rNPOvT/eNZ6rqL6rq2g3njkwzn6yqt67KzBvOva+qRlXtm27Pvs+zhbuqdiV5MMmdSW5Ock9V3TzXet+CbyT5xTHG9yW5Lcm7pjkPJ3lqjHFTkqem26vkPUme3XD715L81jTvvyW5bylTbe23k/zNGONNSX4wi9lXdo+ran+Sdyc5OMZ4c5JdSe7O6u3zw0nuuODYVvt6Z5Kbprf7k3xgh2a80MN56cwfSfLmMcYPJPlMkiNJMj0W707y/dPf+d2pLTvt4bx05lTVDUl+KskXNhyef5/HGLO8Jbk9yYc33D6S5Mhc613Buf9q+g9xMsn107Hrs/hGoqXPN81zIIsH5I8neSJJZfEdW7s32/tlvyV5VZLPZ3oxfMPxVd7j/Um+mOS6LL5R7Ykkb13FfU6yluTExfY1ye8nuWezj1v2zBec+9kkj0zv/59uJPlwkttXZeYkj2XxRORUkn07tc9zXio5f8c/7/R0bGVV1VqStyR5Osl3jzG+lCTTn69d3mQv8f4kv5Tkm9Pt70ry72OMb0y3V22v35DkK0n+aLq88wdV9cqs8B6PMc4k+Y0snkl9Kcm5JMey2vt83lb72uUx+fNJ/np6f2Vnrqq7kpwZY3zyglOzzzxnuGuTYyv7tYdV9Z1J/jzJe8cYX1/2PFupqrcnOTvGOLbx8CYfukp7vTvJrUk+MMZ4S5L/zApdFtnMdF34p5PcmOT1SV6ZxafAF1qlfb6YVb+fpKoeyOLy5SPnD23yYUufuar2JHkgya9sdnqTY1d05jnDfTrJDRtuH0jyLzOud9mq6tuziPYjY4wPTYf/taqun85fn+Tssua7wI8kuauqTiX5kywul7w/ybVVdf5nz6zaXp9OcnqM8fR0+7EsQr6qe5wkP5nk82OMr4wxXkjyoSQ/nNXe5/O22teVfkxW1b1J3p7kHWO6xpDVnfl7s/if+ienx+KBJP9QVa/LDsw8Z7g/nuSm6VX4V2TxAsPjM653WaqqkvxhkmfHGL+54dTjSe6d3r83i2vfSzfGODLGODDGWMtiT/92jPGOJB9N8nPTh63MvEkyxvhyki9W1RunQz+R5J+yons8+UKS26pqz3QfOT/zyu7zBlvt6+NJ3jl91cNtSc6dv6SybFV1R5JfTnLXGOP5DaceT3J3VV1TVTdm8YLf3y9jxo3GGMfHGK8dY6xNj8XTSW6d7uvz7/PMF/PflsUrxJ9L8sAyXlC4hBl/NItPY55J8onp7W1ZXDd+Kslnpz+vW/asm8z+Y0memN5/QxZ36OeS/FmSa5Y93wWz/lCS9Wmf/zLJq1d9j5P8apJPJzmR5I+TXLNq+5zk0Syuwb+QRTzu22pfs/gU/sHp8Xg8i6+YWZWZn8viuvD5x+Dvbfj4B6aZTya5c1VmvuD8qfzvi5Oz77NveQdoxndOAjQj3ADNCDdAM8IN0IxwAzQj3ADNCDdAM/8NHzwchpQeZ0IAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_ready['target_ML_model'].value_counts().plot(kind='barh')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-1    142\n",
       "       14\n",
       "1       3\n",
       "Name: target_ML_model, dtype: int64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ready['target_ML_model'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pubmed_id          object\n",
      "abstract           object\n",
      "target_ML_model    object\n",
      "dtype: object\n"
     ]
    }
   ],
   "source": [
    "print(df_ready.dtypes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ready['abstract'] = df_ready['abstract'].str.lower()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get a word count per sentence column\n",
    "def word_count(sentence):\n",
    "    return len(sentence.split())\n",
    "\n",
    "df_ready['word_count'] = df_ready.abstract.apply(word_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to\n",
      "[nltk_data]     C:\\Users\\iKhan\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n",
      "['i', 'me', 'my', 'myself', 'we', 'our', 'ours', 'ourselves', 'you', \"you're\", \"you've\", \"you'll\", \"you'd\", 'your', 'yours', 'yourself', 'yourselves', 'he', 'him', 'his', 'himself', 'she', \"she's\", 'her', 'hers', 'herself', 'it', \"it's\", 'its', 'itself', 'they', 'them', 'their', 'theirs', 'themselves', 'what', 'which', 'who', 'whom', 'this', 'that', \"that'll\", 'these', 'those', 'am', 'is', 'are', 'was', 'were', 'be', 'been', 'being', 'have', 'has', 'had', 'having', 'do', 'does', 'did', 'doing', 'a', 'an', 'the', 'and', 'but', 'if', 'or', 'because', 'as', 'until', 'while', 'of', 'at', 'by', 'for', 'with', 'about', 'against', 'between', 'into', 'through', 'during', 'before', 'after', 'above', 'below', 'to', 'from', 'up', 'down', 'in', 'out', 'on', 'off', 'over', 'under', 'again', 'further', 'then', 'once', 'here', 'there', 'when', 'where', 'why', 'how', 'all', 'any', 'both', 'each', 'few', 'more', 'most', 'other', 'some', 'such', 'no', 'nor', 'not', 'only', 'own', 'same', 'so', 'than', 'too', 'very', 's', 't', 'can', 'will', 'just', 'don', \"don't\", 'should', \"should've\", 'now', 'd', 'll', 'm', 'o', 're', 've', 'y', 'ain', 'aren', \"aren't\", 'couldn', \"couldn't\", 'didn', \"didn't\", 'doesn', \"doesn't\", 'hadn', \"hadn't\", 'hasn', \"hasn't\", 'haven', \"haven't\", 'isn', \"isn't\", 'ma', 'mightn', \"mightn't\", 'mustn', \"mustn't\", 'needn', \"needn't\", 'shan', \"shan't\", 'shouldn', \"shouldn't\", 'wasn', \"wasn't\", 'weren', \"weren't\", 'won', \"won't\", 'wouldn', \"wouldn't\", 'can', 'do', 'the', 'be', 'will', 'Find', 'third', 'also', 'use', 'etc']\n"
     ]
    }
   ],
   "source": [
    "#data = list(df_ready.abstract)\n",
    "from gensim.utils import simple_preprocess\n",
    "nltk.download('stopwords')\n",
    "stop_words = nltk.corpus.stopwords.words('english')\n",
    "stop_words.extend(['can', 'do', 'the', 'be', 'will', 'Find', 'third', 'also', 'use', 'etc' ])\n",
    "print(stop_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>target_ML_model</th>\n",
       "      <th>word_count</th>\n",
       "      <th>data_words_nostops</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>andrographolide, the major labdane diterpenoid...</td>\n",
       "      <td>-1</td>\n",
       "      <td>136</td>\n",
       "      <td>[andrographolide, major, labdane, diterpenoid,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>dgat2 plays a critical role in hepatic triglyc...</td>\n",
       "      <td>-1</td>\n",
       "      <td>57</td>\n",
       "      <td>[dgat, plays, critical, role, hepatic, triglyc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>despite the success of imatinib at inhibiting ...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[despite, success, imatinib, inhibiting, bcr, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>two benzazaborinine analogues of propranolol w...</td>\n",
       "      <td>-1</td>\n",
       "      <td>66</td>\n",
       "      <td>[two, benzazaborinine, analogues, propranolol,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>our recent study has shown that the natural pr...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[recent, study, shown, natural, product, bouch...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  pubmed_id                                           abstract  \\\n",
       "0  26561364  andrographolide, the major labdane diterpenoid...   \n",
       "1  26561979  dgat2 plays a critical role in hepatic triglyc...   \n",
       "2  26562217  despite the success of imatinib at inhibiting ...   \n",
       "3  26565745  two benzazaborinine analogues of propranolol w...   \n",
       "4  26572282  our recent study has shown that the natural pr...   \n",
       "\n",
       "  target_ML_model  word_count  \\\n",
       "0              -1         136   \n",
       "1              -1          57   \n",
       "2              -1         149   \n",
       "3              -1          66   \n",
       "4              -1         149   \n",
       "\n",
       "                                  data_words_nostops  \n",
       "0  [andrographolide, major, labdane, diterpenoid,...  \n",
       "1  [dgat, plays, critical, role, hepatic, triglyc...  \n",
       "2  [despite, success, imatinib, inhibiting, bcr, ...  \n",
       "3  [two, benzazaborinine, analogues, propranolol,...  \n",
       "4  [recent, study, shown, natural, product, bouch...  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def remove_stopwords(texts):\n",
    "    #return [[word for word in text if word not in stop_words] for doc in texts]\n",
    "    return [[word for word in simple_preprocess(str(doc), deacc=True) if word not in stop_words] for doc in texts]\n",
    "\n",
    "\n",
    "df_ready['data_words_nostops'] = remove_stopwords(df_ready.abstract)\n",
    "df_ready.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ready['data_words_nostops_Joined'] = df_ready.data_words_nostops.apply(','.join)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>target_ML_model</th>\n",
       "      <th>word_count</th>\n",
       "      <th>data_words_nostops</th>\n",
       "      <th>data_words_nostops_Joined</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>andrographolide, the major labdane diterpenoid...</td>\n",
       "      <td>-1</td>\n",
       "      <td>136</td>\n",
       "      <td>[andrographolide, major, labdane, diterpenoid,...</td>\n",
       "      <td>andrographolide,major,labdane,diterpenoid,andr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>dgat2 plays a critical role in hepatic triglyc...</td>\n",
       "      <td>-1</td>\n",
       "      <td>57</td>\n",
       "      <td>[dgat, plays, critical, role, hepatic, triglyc...</td>\n",
       "      <td>dgat,plays,critical,role,hepatic,triglyceride,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>despite the success of imatinib at inhibiting ...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[despite, success, imatinib, inhibiting, bcr, ...</td>\n",
       "      <td>despite,success,imatinib,inhibiting,bcr,abl,tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>two benzazaborinine analogues of propranolol w...</td>\n",
       "      <td>-1</td>\n",
       "      <td>66</td>\n",
       "      <td>[two, benzazaborinine, analogues, propranolol,...</td>\n",
       "      <td>two,benzazaborinine,analogues,propranolol,synt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>our recent study has shown that the natural pr...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[recent, study, shown, natural, product, bouch...</td>\n",
       "      <td>recent,study,shown,natural,product,bouchardati...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  pubmed_id                                           abstract  \\\n",
       "0  26561364  andrographolide, the major labdane diterpenoid...   \n",
       "1  26561979  dgat2 plays a critical role in hepatic triglyc...   \n",
       "2  26562217  despite the success of imatinib at inhibiting ...   \n",
       "3  26565745  two benzazaborinine analogues of propranolol w...   \n",
       "4  26572282  our recent study has shown that the natural pr...   \n",
       "\n",
       "  target_ML_model  word_count  \\\n",
       "0              -1         136   \n",
       "1              -1          57   \n",
       "2              -1         149   \n",
       "3              -1          66   \n",
       "4              -1         149   \n",
       "\n",
       "                                  data_words_nostops  \\\n",
       "0  [andrographolide, major, labdane, diterpenoid,...   \n",
       "1  [dgat, plays, critical, role, hepatic, triglyc...   \n",
       "2  [despite, success, imatinib, inhibiting, bcr, ...   \n",
       "3  [two, benzazaborinine, analogues, propranolol,...   \n",
       "4  [recent, study, shown, natural, product, bouch...   \n",
       "\n",
       "                           data_words_nostops_Joined  \n",
       "0  andrographolide,major,labdane,diterpenoid,andr...  \n",
       "1  dgat,plays,critical,role,hepatic,triglyceride,...  \n",
       "2  despite,success,imatinib,inhibiting,bcr,abl,tr...  \n",
       "3  two,benzazaborinine,analogues,propranolol,synt...  \n",
       "4  recent,study,shown,natural,product,bouchardati...  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ready.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_words_nostops = remove_stopwords(df_ready.abstract)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "from nltk.stem.snowball import SnowballStemmer\n",
    "\n",
    "# Use English stemmer.\n",
    "stemmer = SnowballStemmer(\"english\")\n",
    "\n",
    "df_ready['stemmed'] = df_ready['data_words_nostops'].apply(lambda x: [stemmer.stem(y) for y in x]) # Stem every word.\n",
    "#df_ready.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>target_ML_model</th>\n",
       "      <th>word_count</th>\n",
       "      <th>data_words_nostops</th>\n",
       "      <th>data_words_nostops_Joined</th>\n",
       "      <th>stemmed</th>\n",
       "      <th>stemmed_Joined</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>andrographolide, the major labdane diterpenoid...</td>\n",
       "      <td>-1</td>\n",
       "      <td>136</td>\n",
       "      <td>[andrographolide, major, labdane, diterpenoid,...</td>\n",
       "      <td>andrographolide,major,labdane,diterpenoid,andr...</td>\n",
       "      <td>[andrographolid, major, labdan, diterpenoid, a...</td>\n",
       "      <td>andrographolid,major,labdan,diterpenoid,androg...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>dgat2 plays a critical role in hepatic triglyc...</td>\n",
       "      <td>-1</td>\n",
       "      <td>57</td>\n",
       "      <td>[dgat, plays, critical, role, hepatic, triglyc...</td>\n",
       "      <td>dgat,plays,critical,role,hepatic,triglyceride,...</td>\n",
       "      <td>[dgat, play, critic, role, hepat, triglycerid,...</td>\n",
       "      <td>dgat,play,critic,role,hepat,triglycerid,produc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>despite the success of imatinib at inhibiting ...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[despite, success, imatinib, inhibiting, bcr, ...</td>\n",
       "      <td>despite,success,imatinib,inhibiting,bcr,abl,tr...</td>\n",
       "      <td>[despit, success, imatinib, inhibit, bcr, abl,...</td>\n",
       "      <td>despit,success,imatinib,inhibit,bcr,abl,treat,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>two benzazaborinine analogues of propranolol w...</td>\n",
       "      <td>-1</td>\n",
       "      <td>66</td>\n",
       "      <td>[two, benzazaborinine, analogues, propranolol,...</td>\n",
       "      <td>two,benzazaborinine,analogues,propranolol,synt...</td>\n",
       "      <td>[two, benzazaborinin, analogu, propranolol, sy...</td>\n",
       "      <td>two,benzazaborinin,analogu,propranolol,synthes...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>our recent study has shown that the natural pr...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[recent, study, shown, natural, product, bouch...</td>\n",
       "      <td>recent,study,shown,natural,product,bouchardati...</td>\n",
       "      <td>[recent, studi, shown, natur, product, bouchar...</td>\n",
       "      <td>recent,studi,shown,natur,product,bouchardatin,...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  pubmed_id                                           abstract  \\\n",
       "0  26561364  andrographolide, the major labdane diterpenoid...   \n",
       "1  26561979  dgat2 plays a critical role in hepatic triglyc...   \n",
       "2  26562217  despite the success of imatinib at inhibiting ...   \n",
       "3  26565745  two benzazaborinine analogues of propranolol w...   \n",
       "4  26572282  our recent study has shown that the natural pr...   \n",
       "\n",
       "  target_ML_model  word_count  \\\n",
       "0              -1         136   \n",
       "1              -1          57   \n",
       "2              -1         149   \n",
       "3              -1          66   \n",
       "4              -1         149   \n",
       "\n",
       "                                  data_words_nostops  \\\n",
       "0  [andrographolide, major, labdane, diterpenoid,...   \n",
       "1  [dgat, plays, critical, role, hepatic, triglyc...   \n",
       "2  [despite, success, imatinib, inhibiting, bcr, ...   \n",
       "3  [two, benzazaborinine, analogues, propranolol,...   \n",
       "4  [recent, study, shown, natural, product, bouch...   \n",
       "\n",
       "                           data_words_nostops_Joined  \\\n",
       "0  andrographolide,major,labdane,diterpenoid,andr...   \n",
       "1  dgat,plays,critical,role,hepatic,triglyceride,...   \n",
       "2  despite,success,imatinib,inhibiting,bcr,abl,tr...   \n",
       "3  two,benzazaborinine,analogues,propranolol,synt...   \n",
       "4  recent,study,shown,natural,product,bouchardati...   \n",
       "\n",
       "                                             stemmed  \\\n",
       "0  [andrographolid, major, labdan, diterpenoid, a...   \n",
       "1  [dgat, play, critic, role, hepat, triglycerid,...   \n",
       "2  [despit, success, imatinib, inhibit, bcr, abl,...   \n",
       "3  [two, benzazaborinin, analogu, propranolol, sy...   \n",
       "4  [recent, studi, shown, natur, product, bouchar...   \n",
       "\n",
       "                                      stemmed_Joined  \n",
       "0  andrographolid,major,labdan,diterpenoid,androg...  \n",
       "1  dgat,play,critic,role,hepat,triglycerid,produc...  \n",
       "2  despit,success,imatinib,inhibit,bcr,abl,treat,...  \n",
       "3  two,benzazaborinin,analogu,propranolol,synthes...  \n",
       "4  recent,studi,shown,natur,product,bouchardatin,...  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ready['stemmed_Joined'] = df_ready.stemmed.apply(','.join)\n",
    "df_ready.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>target_ML_model</th>\n",
       "      <th>word_count</th>\n",
       "      <th>data_words_nostops</th>\n",
       "      <th>data_words_nostops_Joined</th>\n",
       "      <th>stemmed</th>\n",
       "      <th>stemmed_Joined</th>\n",
       "      <th>text_lemmatized</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>andrographolide, the major labdane diterpenoid...</td>\n",
       "      <td>-1</td>\n",
       "      <td>136</td>\n",
       "      <td>[andrographolide, major, labdane, diterpenoid,...</td>\n",
       "      <td>andrographolide,major,labdane,diterpenoid,andr...</td>\n",
       "      <td>[andrographolid, major, labdan, diterpenoid, a...</td>\n",
       "      <td>andrographolid,major,labdan,diterpenoid,androg...</td>\n",
       "      <td>[andrographolid,major,labdan,diterpenoid,andro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>dgat2 plays a critical role in hepatic triglyc...</td>\n",
       "      <td>-1</td>\n",
       "      <td>57</td>\n",
       "      <td>[dgat, plays, critical, role, hepatic, triglyc...</td>\n",
       "      <td>dgat,plays,critical,role,hepatic,triglyceride,...</td>\n",
       "      <td>[dgat, play, critic, role, hepat, triglycerid,...</td>\n",
       "      <td>dgat,play,critic,role,hepat,triglycerid,produc...</td>\n",
       "      <td>[dgat,play,critic,role,hepat,triglycerid,produ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>despite the success of imatinib at inhibiting ...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[despite, success, imatinib, inhibiting, bcr, ...</td>\n",
       "      <td>despite,success,imatinib,inhibiting,bcr,abl,tr...</td>\n",
       "      <td>[despit, success, imatinib, inhibit, bcr, abl,...</td>\n",
       "      <td>despit,success,imatinib,inhibit,bcr,abl,treat,...</td>\n",
       "      <td>[despit,success,imatinib,inhibit,bcr,abl,treat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>two benzazaborinine analogues of propranolol w...</td>\n",
       "      <td>-1</td>\n",
       "      <td>66</td>\n",
       "      <td>[two, benzazaborinine, analogues, propranolol,...</td>\n",
       "      <td>two,benzazaborinine,analogues,propranolol,synt...</td>\n",
       "      <td>[two, benzazaborinin, analogu, propranolol, sy...</td>\n",
       "      <td>two,benzazaborinin,analogu,propranolol,synthes...</td>\n",
       "      <td>[two,benzazaborinin,analogu,propranolol,synthe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>our recent study has shown that the natural pr...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[recent, study, shown, natural, product, bouch...</td>\n",
       "      <td>recent,study,shown,natural,product,bouchardati...</td>\n",
       "      <td>[recent, studi, shown, natur, product, bouchar...</td>\n",
       "      <td>recent,studi,shown,natur,product,bouchardatin,...</td>\n",
       "      <td>[recent,studi,shown,natur,product,bouchardatin...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  pubmed_id                                           abstract  \\\n",
       "0  26561364  andrographolide, the major labdane diterpenoid...   \n",
       "1  26561979  dgat2 plays a critical role in hepatic triglyc...   \n",
       "2  26562217  despite the success of imatinib at inhibiting ...   \n",
       "3  26565745  two benzazaborinine analogues of propranolol w...   \n",
       "4  26572282  our recent study has shown that the natural pr...   \n",
       "\n",
       "  target_ML_model  word_count  \\\n",
       "0              -1         136   \n",
       "1              -1          57   \n",
       "2              -1         149   \n",
       "3              -1          66   \n",
       "4              -1         149   \n",
       "\n",
       "                                  data_words_nostops  \\\n",
       "0  [andrographolide, major, labdane, diterpenoid,...   \n",
       "1  [dgat, plays, critical, role, hepatic, triglyc...   \n",
       "2  [despite, success, imatinib, inhibiting, bcr, ...   \n",
       "3  [two, benzazaborinine, analogues, propranolol,...   \n",
       "4  [recent, study, shown, natural, product, bouch...   \n",
       "\n",
       "                           data_words_nostops_Joined  \\\n",
       "0  andrographolide,major,labdane,diterpenoid,andr...   \n",
       "1  dgat,plays,critical,role,hepatic,triglyceride,...   \n",
       "2  despite,success,imatinib,inhibiting,bcr,abl,tr...   \n",
       "3  two,benzazaborinine,analogues,propranolol,synt...   \n",
       "4  recent,study,shown,natural,product,bouchardati...   \n",
       "\n",
       "                                             stemmed  \\\n",
       "0  [andrographolid, major, labdan, diterpenoid, a...   \n",
       "1  [dgat, play, critic, role, hepat, triglycerid,...   \n",
       "2  [despit, success, imatinib, inhibit, bcr, abl,...   \n",
       "3  [two, benzazaborinin, analogu, propranolol, sy...   \n",
       "4  [recent, studi, shown, natur, product, bouchar...   \n",
       "\n",
       "                                      stemmed_Joined  \\\n",
       "0  andrographolid,major,labdan,diterpenoid,androg...   \n",
       "1  dgat,play,critic,role,hepat,triglycerid,produc...   \n",
       "2  despit,success,imatinib,inhibit,bcr,abl,treat,...   \n",
       "3  two,benzazaborinin,analogu,propranolol,synthes...   \n",
       "4  recent,studi,shown,natur,product,bouchardatin,...   \n",
       "\n",
       "                                     text_lemmatized  \n",
       "0  [andrographolid,major,labdan,diterpenoid,andro...  \n",
       "1  [dgat,play,critic,role,hepat,triglycerid,produ...  \n",
       "2  [despit,success,imatinib,inhibit,bcr,abl,treat...  \n",
       "3  [two,benzazaborinin,analogu,propranolol,synthe...  \n",
       "4  [recent,studi,shown,natur,product,bouchardatin...  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import nltk\n",
    "\n",
    "w_tokenizer = nltk.tokenize.WhitespaceTokenizer()\n",
    "lemmatizer = nltk.stem.WordNetLemmatizer()\n",
    "\n",
    "def lemmatize_text(text):\n",
    "    return [lemmatizer.lemmatize(w) for w in w_tokenizer.tokenize(text)]\n",
    "\n",
    "df_ready['text_lemmatized'] = df_ready.stemmed_Joined.apply(lemmatize_text)\n",
    "df_ready.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ready['text_lemmatized_Joined'] = df_ready.text_lemmatized.apply(','.join)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ready['bigrams'] = df_ready['text_lemmatized_Joined'].apply(lambda row: list(nltk.bigrams(row.split(','))))\n",
    "#df_ready['bigrams'] = df_ready['text_lemmatized'].apply(lambda row: list(nltk.bigrams(row.split(','))))\n",
    "#df_ready['trigrams'] = df_ready['data_words_nostops_Joined'].apply(lambda row: list(nltk.trigrams(row.split(' '))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>target_ML_model</th>\n",
       "      <th>word_count</th>\n",
       "      <th>data_words_nostops</th>\n",
       "      <th>data_words_nostops_Joined</th>\n",
       "      <th>stemmed</th>\n",
       "      <th>stemmed_Joined</th>\n",
       "      <th>text_lemmatized</th>\n",
       "      <th>text_lemmatized_Joined</th>\n",
       "      <th>bigrams</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>andrographolide, the major labdane diterpenoid...</td>\n",
       "      <td>-1</td>\n",
       "      <td>136</td>\n",
       "      <td>[andrographolide, major, labdane, diterpenoid,...</td>\n",
       "      <td>andrographolide,major,labdane,diterpenoid,andr...</td>\n",
       "      <td>[andrographolid, major, labdan, diterpenoid, a...</td>\n",
       "      <td>andrographolid,major,labdan,diterpenoid,androg...</td>\n",
       "      <td>[andrographolid,major,labdan,diterpenoid,andro...</td>\n",
       "      <td>andrographolid,major,labdan,diterpenoid,androg...</td>\n",
       "      <td>[(andrographolid, major), (major, labdan), (la...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>dgat2 plays a critical role in hepatic triglyc...</td>\n",
       "      <td>-1</td>\n",
       "      <td>57</td>\n",
       "      <td>[dgat, plays, critical, role, hepatic, triglyc...</td>\n",
       "      <td>dgat,plays,critical,role,hepatic,triglyceride,...</td>\n",
       "      <td>[dgat, play, critic, role, hepat, triglycerid,...</td>\n",
       "      <td>dgat,play,critic,role,hepat,triglycerid,produc...</td>\n",
       "      <td>[dgat,play,critic,role,hepat,triglycerid,produ...</td>\n",
       "      <td>dgat,play,critic,role,hepat,triglycerid,produc...</td>\n",
       "      <td>[(dgat, play), (play, critic), (critic, role),...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>despite the success of imatinib at inhibiting ...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[despite, success, imatinib, inhibiting, bcr, ...</td>\n",
       "      <td>despite,success,imatinib,inhibiting,bcr,abl,tr...</td>\n",
       "      <td>[despit, success, imatinib, inhibit, bcr, abl,...</td>\n",
       "      <td>despit,success,imatinib,inhibit,bcr,abl,treat,...</td>\n",
       "      <td>[despit,success,imatinib,inhibit,bcr,abl,treat...</td>\n",
       "      <td>despit,success,imatinib,inhibit,bcr,abl,treat,...</td>\n",
       "      <td>[(despit, success), (success, imatinib), (imat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>two benzazaborinine analogues of propranolol w...</td>\n",
       "      <td>-1</td>\n",
       "      <td>66</td>\n",
       "      <td>[two, benzazaborinine, analogues, propranolol,...</td>\n",
       "      <td>two,benzazaborinine,analogues,propranolol,synt...</td>\n",
       "      <td>[two, benzazaborinin, analogu, propranolol, sy...</td>\n",
       "      <td>two,benzazaborinin,analogu,propranolol,synthes...</td>\n",
       "      <td>[two,benzazaborinin,analogu,propranolol,synthe...</td>\n",
       "      <td>two,benzazaborinin,analogu,propranolol,synthes...</td>\n",
       "      <td>[(two, benzazaborinin), (benzazaborinin, analo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>our recent study has shown that the natural pr...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[recent, study, shown, natural, product, bouch...</td>\n",
       "      <td>recent,study,shown,natural,product,bouchardati...</td>\n",
       "      <td>[recent, studi, shown, natur, product, bouchar...</td>\n",
       "      <td>recent,studi,shown,natur,product,bouchardatin,...</td>\n",
       "      <td>[recent,studi,shown,natur,product,bouchardatin...</td>\n",
       "      <td>recent,studi,shown,natur,product,bouchardatin,...</td>\n",
       "      <td>[(recent, studi), (studi, shown), (shown, natu...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  pubmed_id                                           abstract  \\\n",
       "0  26561364  andrographolide, the major labdane diterpenoid...   \n",
       "1  26561979  dgat2 plays a critical role in hepatic triglyc...   \n",
       "2  26562217  despite the success of imatinib at inhibiting ...   \n",
       "3  26565745  two benzazaborinine analogues of propranolol w...   \n",
       "4  26572282  our recent study has shown that the natural pr...   \n",
       "\n",
       "  target_ML_model  word_count  \\\n",
       "0              -1         136   \n",
       "1              -1          57   \n",
       "2              -1         149   \n",
       "3              -1          66   \n",
       "4              -1         149   \n",
       "\n",
       "                                  data_words_nostops  \\\n",
       "0  [andrographolide, major, labdane, diterpenoid,...   \n",
       "1  [dgat, plays, critical, role, hepatic, triglyc...   \n",
       "2  [despite, success, imatinib, inhibiting, bcr, ...   \n",
       "3  [two, benzazaborinine, analogues, propranolol,...   \n",
       "4  [recent, study, shown, natural, product, bouch...   \n",
       "\n",
       "                           data_words_nostops_Joined  \\\n",
       "0  andrographolide,major,labdane,diterpenoid,andr...   \n",
       "1  dgat,plays,critical,role,hepatic,triglyceride,...   \n",
       "2  despite,success,imatinib,inhibiting,bcr,abl,tr...   \n",
       "3  two,benzazaborinine,analogues,propranolol,synt...   \n",
       "4  recent,study,shown,natural,product,bouchardati...   \n",
       "\n",
       "                                             stemmed  \\\n",
       "0  [andrographolid, major, labdan, diterpenoid, a...   \n",
       "1  [dgat, play, critic, role, hepat, triglycerid,...   \n",
       "2  [despit, success, imatinib, inhibit, bcr, abl,...   \n",
       "3  [two, benzazaborinin, analogu, propranolol, sy...   \n",
       "4  [recent, studi, shown, natur, product, bouchar...   \n",
       "\n",
       "                                      stemmed_Joined  \\\n",
       "0  andrographolid,major,labdan,diterpenoid,androg...   \n",
       "1  dgat,play,critic,role,hepat,triglycerid,produc...   \n",
       "2  despit,success,imatinib,inhibit,bcr,abl,treat,...   \n",
       "3  two,benzazaborinin,analogu,propranolol,synthes...   \n",
       "4  recent,studi,shown,natur,product,bouchardatin,...   \n",
       "\n",
       "                                     text_lemmatized  \\\n",
       "0  [andrographolid,major,labdan,diterpenoid,andro...   \n",
       "1  [dgat,play,critic,role,hepat,triglycerid,produ...   \n",
       "2  [despit,success,imatinib,inhibit,bcr,abl,treat...   \n",
       "3  [two,benzazaborinin,analogu,propranolol,synthe...   \n",
       "4  [recent,studi,shown,natur,product,bouchardatin...   \n",
       "\n",
       "                              text_lemmatized_Joined  \\\n",
       "0  andrographolid,major,labdan,diterpenoid,androg...   \n",
       "1  dgat,play,critic,role,hepat,triglycerid,produc...   \n",
       "2  despit,success,imatinib,inhibit,bcr,abl,treat,...   \n",
       "3  two,benzazaborinin,analogu,propranolol,synthes...   \n",
       "4  recent,studi,shown,natur,product,bouchardatin,...   \n",
       "\n",
       "                                             bigrams  \n",
       "0  [(andrographolid, major), (major, labdan), (la...  \n",
       "1  [(dgat, play), (play, critic), (critic, role),...  \n",
       "2  [(despit, success), (success, imatinib), (imat...  \n",
       "3  [(two, benzazaborinin), (benzazaborinin, analo...  \n",
       "4  [(recent, studi), (studi, shown), (shown, natu...  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ready.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import MultiLabelBinarizer\n",
    "\n",
    "mlb = MultiLabelBinarizer()\n",
    "df_ready = df_ready.join(pd.DataFrame(mlb.fit_transform(df_ready.pop('bigrams')),\n",
    "                          columns=mlb.classes_,\n",
    "                          index=df_ready.index))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>target_ML_model</th>\n",
       "      <th>word_count</th>\n",
       "      <th>data_words_nostops</th>\n",
       "      <th>data_words_nostops_Joined</th>\n",
       "      <th>stemmed</th>\n",
       "      <th>stemmed_Joined</th>\n",
       "      <th>text_lemmatized</th>\n",
       "      <th>text_lemmatized_Joined</th>\n",
       "      <th>...</th>\n",
       "      <th>(ylmethylen, hex)</th>\n",
       "      <th>(zinc, chelat)</th>\n",
       "      <th>(zinc, finger)</th>\n",
       "      <th>(zip, code)</th>\n",
       "      <th>(zone, express)</th>\n",
       "      <th>(zone, misexpress)</th>\n",
       "      <th>(zwitterion, guanidium)</th>\n",
       "      <th>(zygot, contribut)</th>\n",
       "      <th>(zygot, express)</th>\n",
       "      <th>(zygot, rna)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>andrographolide, the major labdane diterpenoid...</td>\n",
       "      <td>-1</td>\n",
       "      <td>136</td>\n",
       "      <td>[andrographolide, major, labdane, diterpenoid,...</td>\n",
       "      <td>andrographolide,major,labdane,diterpenoid,andr...</td>\n",
       "      <td>[andrographolid, major, labdan, diterpenoid, a...</td>\n",
       "      <td>andrographolid,major,labdan,diterpenoid,androg...</td>\n",
       "      <td>[andrographolid,major,labdan,diterpenoid,andro...</td>\n",
       "      <td>andrographolid,major,labdan,diterpenoid,androg...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>dgat2 plays a critical role in hepatic triglyc...</td>\n",
       "      <td>-1</td>\n",
       "      <td>57</td>\n",
       "      <td>[dgat, plays, critical, role, hepatic, triglyc...</td>\n",
       "      <td>dgat,plays,critical,role,hepatic,triglyceride,...</td>\n",
       "      <td>[dgat, play, critic, role, hepat, triglycerid,...</td>\n",
       "      <td>dgat,play,critic,role,hepat,triglycerid,produc...</td>\n",
       "      <td>[dgat,play,critic,role,hepat,triglycerid,produ...</td>\n",
       "      <td>dgat,play,critic,role,hepat,triglycerid,produc...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>despite the success of imatinib at inhibiting ...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[despite, success, imatinib, inhibiting, bcr, ...</td>\n",
       "      <td>despite,success,imatinib,inhibiting,bcr,abl,tr...</td>\n",
       "      <td>[despit, success, imatinib, inhibit, bcr, abl,...</td>\n",
       "      <td>despit,success,imatinib,inhibit,bcr,abl,treat,...</td>\n",
       "      <td>[despit,success,imatinib,inhibit,bcr,abl,treat...</td>\n",
       "      <td>despit,success,imatinib,inhibit,bcr,abl,treat,...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>two benzazaborinine analogues of propranolol w...</td>\n",
       "      <td>-1</td>\n",
       "      <td>66</td>\n",
       "      <td>[two, benzazaborinine, analogues, propranolol,...</td>\n",
       "      <td>two,benzazaborinine,analogues,propranolol,synt...</td>\n",
       "      <td>[two, benzazaborinin, analogu, propranolol, sy...</td>\n",
       "      <td>two,benzazaborinin,analogu,propranolol,synthes...</td>\n",
       "      <td>[two,benzazaborinin,analogu,propranolol,synthe...</td>\n",
       "      <td>two,benzazaborinin,analogu,propranolol,synthes...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>our recent study has shown that the natural pr...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[recent, study, shown, natural, product, bouch...</td>\n",
       "      <td>recent,study,shown,natural,product,bouchardati...</td>\n",
       "      <td>[recent, studi, shown, natur, product, bouchar...</td>\n",
       "      <td>recent,studi,shown,natur,product,bouchardatin,...</td>\n",
       "      <td>[recent,studi,shown,natur,product,bouchardatin...</td>\n",
       "      <td>recent,studi,shown,natur,product,bouchardatin,...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 9913 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  pubmed_id                                           abstract  \\\n",
       "0  26561364  andrographolide, the major labdane diterpenoid...   \n",
       "1  26561979  dgat2 plays a critical role in hepatic triglyc...   \n",
       "2  26562217  despite the success of imatinib at inhibiting ...   \n",
       "3  26565745  two benzazaborinine analogues of propranolol w...   \n",
       "4  26572282  our recent study has shown that the natural pr...   \n",
       "\n",
       "  target_ML_model  word_count  \\\n",
       "0              -1         136   \n",
       "1              -1          57   \n",
       "2              -1         149   \n",
       "3              -1          66   \n",
       "4              -1         149   \n",
       "\n",
       "                                  data_words_nostops  \\\n",
       "0  [andrographolide, major, labdane, diterpenoid,...   \n",
       "1  [dgat, plays, critical, role, hepatic, triglyc...   \n",
       "2  [despite, success, imatinib, inhibiting, bcr, ...   \n",
       "3  [two, benzazaborinine, analogues, propranolol,...   \n",
       "4  [recent, study, shown, natural, product, bouch...   \n",
       "\n",
       "                           data_words_nostops_Joined  \\\n",
       "0  andrographolide,major,labdane,diterpenoid,andr...   \n",
       "1  dgat,plays,critical,role,hepatic,triglyceride,...   \n",
       "2  despite,success,imatinib,inhibiting,bcr,abl,tr...   \n",
       "3  two,benzazaborinine,analogues,propranolol,synt...   \n",
       "4  recent,study,shown,natural,product,bouchardati...   \n",
       "\n",
       "                                             stemmed  \\\n",
       "0  [andrographolid, major, labdan, diterpenoid, a...   \n",
       "1  [dgat, play, critic, role, hepat, triglycerid,...   \n",
       "2  [despit, success, imatinib, inhibit, bcr, abl,...   \n",
       "3  [two, benzazaborinin, analogu, propranolol, sy...   \n",
       "4  [recent, studi, shown, natur, product, bouchar...   \n",
       "\n",
       "                                      stemmed_Joined  \\\n",
       "0  andrographolid,major,labdan,diterpenoid,androg...   \n",
       "1  dgat,play,critic,role,hepat,triglycerid,produc...   \n",
       "2  despit,success,imatinib,inhibit,bcr,abl,treat,...   \n",
       "3  two,benzazaborinin,analogu,propranolol,synthes...   \n",
       "4  recent,studi,shown,natur,product,bouchardatin,...   \n",
       "\n",
       "                                     text_lemmatized  \\\n",
       "0  [andrographolid,major,labdan,diterpenoid,andro...   \n",
       "1  [dgat,play,critic,role,hepat,triglycerid,produ...   \n",
       "2  [despit,success,imatinib,inhibit,bcr,abl,treat...   \n",
       "3  [two,benzazaborinin,analogu,propranolol,synthe...   \n",
       "4  [recent,studi,shown,natur,product,bouchardatin...   \n",
       "\n",
       "                              text_lemmatized_Joined  ...  (ylmethylen, hex)  \\\n",
       "0  andrographolid,major,labdan,diterpenoid,androg...  ...                  0   \n",
       "1  dgat,play,critic,role,hepat,triglycerid,produc...  ...                  0   \n",
       "2  despit,success,imatinib,inhibit,bcr,abl,treat,...  ...                  0   \n",
       "3  two,benzazaborinin,analogu,propranolol,synthes...  ...                  0   \n",
       "4  recent,studi,shown,natur,product,bouchardatin,...  ...                  0   \n",
       "\n",
       "   (zinc, chelat)  (zinc, finger)  (zip, code)  (zone, express)  \\\n",
       "0               0               0            0                0   \n",
       "1               0               0            0                0   \n",
       "2               0               0            0                0   \n",
       "3               0               0            0                0   \n",
       "4               0               0            0                0   \n",
       "\n",
       "   (zone, misexpress)  (zwitterion, guanidium)  (zygot, contribut)  \\\n",
       "0                   0                        0                   0   \n",
       "1                   0                        0                   0   \n",
       "2                   0                        0                   0   \n",
       "3                   0                        0                   0   \n",
       "4                   0                        0                   0   \n",
       "\n",
       "   (zygot, express)  (zygot, rna)  \n",
       "0                 0             0  \n",
       "1                 0             0  \n",
       "2                 0             0  \n",
       "3                 0             0  \n",
       "4                 0             0  \n",
       "\n",
       "[5 rows x 9913 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ready.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(159, 9913)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ready.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['aaic', 'ab', 'abietan', 'abil', 'abl', 'abnorm', 'absenc', 'absolut',\n",
       "       'abus', 'ac',\n",
       "       ...\n",
       "       'yeast', 'yet', 'yield', 'yl', 'ylmethylen', 'zinc', 'zip', 'zone',\n",
       "       'zwitterion', 'zygot'],\n",
       "      dtype='object', length=2766)"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df_ready['data_words_nostops_Joined'].apply(pd.Series)\n",
    "mono_data = df_ready.text_lemmatized.apply(','.join).str.get_dummies(sep=',')\n",
    "mono_data.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#full_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(159, 12679)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_data = pd.concat([df_ready, mono_data], axis=1, sort=False)\n",
    "full_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([                'pubmed_id',                  'abstract',\n",
       "                 'target_ML_model',                'word_count',\n",
       "              'data_words_nostops', 'data_words_nostops_Joined',\n",
       "                         'stemmed',            'stemmed_Joined',\n",
       "                 'text_lemmatized',    'text_lemmatized_Joined',\n",
       "       ...\n",
       "                           'yeast',                       'yet',\n",
       "                           'yield',                        'yl',\n",
       "                      'ylmethylen',                      'zinc',\n",
       "                             'zip',                      'zone',\n",
       "                      'zwitterion',                     'zygot'],\n",
       "      dtype='object', length=12679)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_data.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "#mono_data[mono_data.columns[pd.Series(mono_data.columns).str.startswith('abstract')]]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dtype('int64')"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_data['class'].dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>abstract</th>\n",
       "      <th>target_ML_model</th>\n",
       "      <th>word_count</th>\n",
       "      <th>data_words_nostops</th>\n",
       "      <th>data_words_nostops_Joined</th>\n",
       "      <th>stemmed</th>\n",
       "      <th>stemmed_Joined</th>\n",
       "      <th>text_lemmatized</th>\n",
       "      <th>text_lemmatized_Joined</th>\n",
       "      <th>...</th>\n",
       "      <th>yeast</th>\n",
       "      <th>yet</th>\n",
       "      <th>yield</th>\n",
       "      <th>yl</th>\n",
       "      <th>ylmethylen</th>\n",
       "      <th>zinc</th>\n",
       "      <th>zip</th>\n",
       "      <th>zone</th>\n",
       "      <th>zwitterion</th>\n",
       "      <th>zygot</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>andrographolide, the major labdane diterpenoid...</td>\n",
       "      <td>-1</td>\n",
       "      <td>136</td>\n",
       "      <td>[andrographolide, major, labdane, diterpenoid,...</td>\n",
       "      <td>andrographolide,major,labdane,diterpenoid,andr...</td>\n",
       "      <td>[andrographolid, major, labdan, diterpenoid, a...</td>\n",
       "      <td>andrographolid,major,labdan,diterpenoid,androg...</td>\n",
       "      <td>[andrographolid,major,labdan,diterpenoid,andro...</td>\n",
       "      <td>andrographolid,major,labdan,diterpenoid,androg...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>dgat2 plays a critical role in hepatic triglyc...</td>\n",
       "      <td>-1</td>\n",
       "      <td>57</td>\n",
       "      <td>[dgat, plays, critical, role, hepatic, triglyc...</td>\n",
       "      <td>dgat,plays,critical,role,hepatic,triglyceride,...</td>\n",
       "      <td>[dgat, play, critic, role, hepat, triglycerid,...</td>\n",
       "      <td>dgat,play,critic,role,hepat,triglycerid,produc...</td>\n",
       "      <td>[dgat,play,critic,role,hepat,triglycerid,produ...</td>\n",
       "      <td>dgat,play,critic,role,hepat,triglycerid,produc...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>despite the success of imatinib at inhibiting ...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[despite, success, imatinib, inhibiting, bcr, ...</td>\n",
       "      <td>despite,success,imatinib,inhibiting,bcr,abl,tr...</td>\n",
       "      <td>[despit, success, imatinib, inhibit, bcr, abl,...</td>\n",
       "      <td>despit,success,imatinib,inhibit,bcr,abl,treat,...</td>\n",
       "      <td>[despit,success,imatinib,inhibit,bcr,abl,treat...</td>\n",
       "      <td>despit,success,imatinib,inhibit,bcr,abl,treat,...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>two benzazaborinine analogues of propranolol w...</td>\n",
       "      <td>-1</td>\n",
       "      <td>66</td>\n",
       "      <td>[two, benzazaborinine, analogues, propranolol,...</td>\n",
       "      <td>two,benzazaborinine,analogues,propranolol,synt...</td>\n",
       "      <td>[two, benzazaborinin, analogu, propranolol, sy...</td>\n",
       "      <td>two,benzazaborinin,analogu,propranolol,synthes...</td>\n",
       "      <td>[two,benzazaborinin,analogu,propranolol,synthe...</td>\n",
       "      <td>two,benzazaborinin,analogu,propranolol,synthes...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>our recent study has shown that the natural pr...</td>\n",
       "      <td>-1</td>\n",
       "      <td>149</td>\n",
       "      <td>[recent, study, shown, natural, product, bouch...</td>\n",
       "      <td>recent,study,shown,natural,product,bouchardati...</td>\n",
       "      <td>[recent, studi, shown, natur, product, bouchar...</td>\n",
       "      <td>recent,studi,shown,natur,product,bouchardatin,...</td>\n",
       "      <td>[recent,studi,shown,natur,product,bouchardatin...</td>\n",
       "      <td>recent,studi,shown,natur,product,bouchardatin,...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 12679 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  pubmed_id                                           abstract  \\\n",
       "0  26561364  andrographolide, the major labdane diterpenoid...   \n",
       "1  26561979  dgat2 plays a critical role in hepatic triglyc...   \n",
       "2  26562217  despite the success of imatinib at inhibiting ...   \n",
       "3  26565745  two benzazaborinine analogues of propranolol w...   \n",
       "4  26572282  our recent study has shown that the natural pr...   \n",
       "\n",
       "  target_ML_model  word_count  \\\n",
       "0              -1         136   \n",
       "1              -1          57   \n",
       "2              -1         149   \n",
       "3              -1          66   \n",
       "4              -1         149   \n",
       "\n",
       "                                  data_words_nostops  \\\n",
       "0  [andrographolide, major, labdane, diterpenoid,...   \n",
       "1  [dgat, plays, critical, role, hepatic, triglyc...   \n",
       "2  [despite, success, imatinib, inhibiting, bcr, ...   \n",
       "3  [two, benzazaborinine, analogues, propranolol,...   \n",
       "4  [recent, study, shown, natural, product, bouch...   \n",
       "\n",
       "                           data_words_nostops_Joined  \\\n",
       "0  andrographolide,major,labdane,diterpenoid,andr...   \n",
       "1  dgat,plays,critical,role,hepatic,triglyceride,...   \n",
       "2  despite,success,imatinib,inhibiting,bcr,abl,tr...   \n",
       "3  two,benzazaborinine,analogues,propranolol,synt...   \n",
       "4  recent,study,shown,natural,product,bouchardati...   \n",
       "\n",
       "                                             stemmed  \\\n",
       "0  [andrographolid, major, labdan, diterpenoid, a...   \n",
       "1  [dgat, play, critic, role, hepat, triglycerid,...   \n",
       "2  [despit, success, imatinib, inhibit, bcr, abl,...   \n",
       "3  [two, benzazaborinin, analogu, propranolol, sy...   \n",
       "4  [recent, studi, shown, natur, product, bouchar...   \n",
       "\n",
       "                                      stemmed_Joined  \\\n",
       "0  andrographolid,major,labdan,diterpenoid,androg...   \n",
       "1  dgat,play,critic,role,hepat,triglycerid,produc...   \n",
       "2  despit,success,imatinib,inhibit,bcr,abl,treat,...   \n",
       "3  two,benzazaborinin,analogu,propranolol,synthes...   \n",
       "4  recent,studi,shown,natur,product,bouchardatin,...   \n",
       "\n",
       "                                     text_lemmatized  \\\n",
       "0  [andrographolid,major,labdan,diterpenoid,andro...   \n",
       "1  [dgat,play,critic,role,hepat,triglycerid,produ...   \n",
       "2  [despit,success,imatinib,inhibit,bcr,abl,treat...   \n",
       "3  [two,benzazaborinin,analogu,propranolol,synthe...   \n",
       "4  [recent,studi,shown,natur,product,bouchardatin...   \n",
       "\n",
       "                              text_lemmatized_Joined  ...  yeast  yet  yield  \\\n",
       "0  andrographolid,major,labdan,diterpenoid,androg...  ...      0    0      0   \n",
       "1  dgat,play,critic,role,hepat,triglycerid,produc...  ...      0    0      0   \n",
       "2  despit,success,imatinib,inhibit,bcr,abl,treat,...  ...      0    0      0   \n",
       "3  two,benzazaborinin,analogu,propranolol,synthes...  ...      0    0      0   \n",
       "4  recent,studi,shown,natur,product,bouchardatin,...  ...      0    0      0   \n",
       "\n",
       "   yl  ylmethylen  zinc  zip  zone  zwitterion  zygot  \n",
       "0   0           0     0    0     0           0      0  \n",
       "1   0           0     0    0     0           0      0  \n",
       "2   0           0     0    0     0           0      0  \n",
       "3   0           0     0    0     0           0      0  \n",
       "4   0           0     0    0     0           0      0  \n",
       "\n",
       "[5 rows x 12679 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "#from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.linear_model import Lasso, LogisticRegression\n",
    "from sklearn.metrics import classification_report, confusion_matrix\n",
    "from sklearn.model_selection import train_test_split\n",
    "#x = df_ready.values#sentences = df_ready['data_lemmatized_bigrams_Joined'].values\n",
    "#X = df_ready.values#sentences = df_ready['data_lemmatized_bigrams_Joined'].values\n",
    "#full_data  = full_data.set_index('pubmed_id')\n",
    "#X = df_ready.loc[:, df_ready.columns != ['abstract', 'word_count', 'data_words_nostops', 'data_words_nostops_Joined']]\n",
    "\n",
    "X = full_data.drop(['abstract', 'word_count', 'data_words_nostops', 'data_words_nostops_Joined','text_lemmatized', 'text_lemmatized_Joined','target_ML_model', 'stemmed', 'stemmed_Joined'], axis=1)\n",
    "#full_data['target_ML_model'] = full_data['target_ML_model'].astype(float)\n",
    "\n",
    "y = full_data['target_ML_model'].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>(aaic, herein)</th>\n",
       "      <th>(ab, cultur)</th>\n",
       "      <th>(abietan, diterpenoid)</th>\n",
       "      <th>(abietan, skeleton)</th>\n",
       "      <th>(abil, alter)</th>\n",
       "      <th>(abil, form)</th>\n",
       "      <th>(abil, hexahydro)</th>\n",
       "      <th>(abil, induc)</th>\n",
       "      <th>(abil, inhibit)</th>\n",
       "      <th>...</th>\n",
       "      <th>yeast</th>\n",
       "      <th>yet</th>\n",
       "      <th>yield</th>\n",
       "      <th>yl</th>\n",
       "      <th>ylmethylen</th>\n",
       "      <th>zinc</th>\n",
       "      <th>zip</th>\n",
       "      <th>zone</th>\n",
       "      <th>zwitterion</th>\n",
       "      <th>zygot</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26561364</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26561979</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26562217</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26565745</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26572282</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 12670 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  pubmed_id  (aaic, herein)  (ab, cultur)  (abietan, diterpenoid)  \\\n",
       "0  26561364               0             0                       0   \n",
       "1  26561979               0             0                       0   \n",
       "2  26562217               0             0                       0   \n",
       "3  26565745               0             0                       0   \n",
       "4  26572282               0             0                       0   \n",
       "\n",
       "   (abietan, skeleton)  (abil, alter)  (abil, form)  (abil, hexahydro)  \\\n",
       "0                    0              0             0                  0   \n",
       "1                    0              0             0                  0   \n",
       "2                    0              0             0                  0   \n",
       "3                    0              0             0                  0   \n",
       "4                    0              0             0                  0   \n",
       "\n",
       "   (abil, induc)  (abil, inhibit)  ...  yeast  yet  yield  yl  ylmethylen  \\\n",
       "0              0                0  ...      0    0      0   0           0   \n",
       "1              0                0  ...      0    0      0   0           0   \n",
       "2              0                0  ...      0    0      0   0           0   \n",
       "3              0                0  ...      0    0      0   0           0   \n",
       "4              0                0  ...      0    0      0   0           0   \n",
       "\n",
       "   zinc  zip  zone  zwitterion  zygot  \n",
       "0     0    0     0           0      0  \n",
       "1     0    0     0           0      0  \n",
       "2     0    0     0           0      0  \n",
       "3     0    0     0           0      0  \n",
       "4     0    0     0           0      0  \n",
       "\n",
       "[5 rows x 12670 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "#X.get_dtype_counts() \n",
    "X  = X.set_index('pubmed_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(119, 12669)\n",
      "(40, 12669)\n",
      "(119,)\n",
      "(40,)\n"
     ]
    }
   ],
   "source": [
    "x_train, x_test, y_train, y_test = train_test_split(\n",
    "X, y, test_size=0.25, random_state=1000)\n",
    "#y = df_ready['class'].values\n",
    "print(x_train.shape)\n",
    "print(x_test.shape)\n",
    "print(y_train.shape)\n",
    "print(y_test.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(aaic, herein)            int32\n",
       "(ab, cultur)              int32\n",
       "(abietan, diterpenoid)    int32\n",
       "(abietan, skeleton)       int32\n",
       "(abil, alter)             int32\n",
       "                          ...  \n",
       "zinc                      int64\n",
       "zip                       int64\n",
       "zone                      int64\n",
       "zwitterion                int64\n",
       "zygot                     int64\n",
       "Length: 12669, dtype: object"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x_train.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.\n",
      "  FutureWarning)\n",
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:460: FutureWarning: Default multi_class will be changed to 'auto' in 0.22. Specify the multi_class option to silence this warning.\n",
      "  \"this warning.\", FutureWarning)\n",
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.\n",
      "  FutureWarning)\n",
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:460: FutureWarning: Default multi_class will be changed to 'auto' in 0.22. Specify the multi_class option to silence this warning.\n",
      "  \"this warning.\", FutureWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total features: 12669\n",
      "selected features: 3877\n",
      "features with coefficients shrank to zero: 8409\n",
      "total features: 12669\n",
      "selected features: 3895\n",
      "features with coefficients shrank to zero: 8409\n",
      "total features: 12669\n",
      "selected features: 4012\n",
      "features with coefficients shrank to zero: 8409\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.\n",
      "  FutureWarning)\n",
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:460: FutureWarning: Default multi_class will be changed to 'auto' in 0.22. Specify the multi_class option to silence this warning.\n",
      "  \"this warning.\", FutureWarning)\n",
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.\n",
      "  FutureWarning)\n",
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:460: FutureWarning: Default multi_class will be changed to 'auto' in 0.22. Specify the multi_class option to silence this warning.\n",
      "  \"this warning.\", FutureWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total features: 12669\n",
      "selected features: 2482\n",
      "features with coefficients shrank to zero: 8409\n",
      "total features: 12669\n",
      "selected features: 1358\n",
      "features with coefficients shrank to zero: 8409\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.\n",
      "  FutureWarning)\n",
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:460: FutureWarning: Default multi_class will be changed to 'auto' in 0.22. Specify the multi_class option to silence this warning.\n",
      "  \"this warning.\", FutureWarning)\n"
     ]
    }
   ],
   "source": [
    "C = [10, 1, .1, .001, .0001]\n",
    "\n",
    "for c in C:\n",
    "    sel_ = SelectFromModel(LogisticRegression(C=c, penalty='l2'))\n",
    "    sel_.fit(x_train, y_train)\n",
    "    selected_features = sel_.get_support()\n",
    "    np.sum(selected_features)\n",
    "    selected_feat = x_train.columns[(sel_.get_support())]\n",
    "    print('total features: {}'.format((x_train.shape[1])))\n",
    "    print('selected features: {}'.format(len(selected_feat)))\n",
    "    print('features with coefficients shrank to zero: {}'.format(\n",
    "          np.sum(sel_.estimator_.coef_ == 0)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\nsel_ = SelectFromModel(LogisticRegression(C=0.0001, penalty='l2'))\\nsel_.fit(x_train, y_train)\\nselected_features = sel_.get_support()\\nnp.sum(selected_features)\\n\\nselected_feat = x_train.columns[(sel_.get_support())]\\nprint('total features: {}'.format((x_train.shape[1])))\\nprint('selected features: {}'.format(len(selected_feat)))\\nprint('features with coefficients shrank to zero: {}'.format(\\n      np.sum(sel_.estimator_.coef_ == 0)))\\n\\nC = [10, 1, .1, .001]\\n\\nfor c in C:\\n    clf = LogisticRegression(penalty='l2', C=c, solver='liblinear')\\n    clf.fit(x_train, y_train)\\n    print('C:', c)\\n    print('Coefficient of each feature:', clf.coef_)\\n    print('Training accuracy:', clf.score(x_train, y_train))\\n    print('Test accuracy:', clf.score(x_test, y_test))\\n    print(classification_report(y_test, clf.predict(x_test)))\\n    #sel_ = SelectFromModel(LogisticRegression(C=c, penalty='l1'))\\n    #sel_.get_support()\\n    print('')\""
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "sel_ = SelectFromModel(LogisticRegression(C=0.0001, penalty='l2'))\n",
    "sel_.fit(x_train, y_train)\n",
    "selected_features = sel_.get_support()\n",
    "np.sum(selected_features)\n",
    "\n",
    "selected_feat = x_train.columns[(sel_.get_support())]\n",
    "print('total features: {}'.format((x_train.shape[1])))\n",
    "print('selected features: {}'.format(len(selected_feat)))\n",
    "print('features with coefficients shrank to zero: {}'.format(\n",
    "      np.sum(sel_.estimator_.coef_ == 0)))\n",
    "\n",
    "C = [10, 1, .1, .001]\n",
    "\n",
    "for c in C:\n",
    "    clf = LogisticRegression(penalty='l2', C=c, solver='liblinear')\n",
    "    clf.fit(x_train, y_train)\n",
    "    print('C:', c)\n",
    "    print('Coefficient of each feature:', clf.coef_)\n",
    "    print('Training accuracy:', clf.score(x_train, y_train))\n",
    "    print('Test accuracy:', clf.score(x_test, y_test))\n",
    "    print(classification_report(y_test, clf.predict(x_test)))\n",
    "    #sel_ = SelectFromModel(LogisticRegression(C=c, penalty='l1'))\n",
    "    #sel_.get_support()\n",
    "    print('')\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "#(model.predict_proba(x_test)[:,1] > threshold).astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C: 0.0001\n",
      "Coefficient of each feature: [[-9.75196861e-05 -4.41332754e-04 -9.82474291e-05 ...  1.01033810e-04\n",
      "  -3.43355446e-04 -3.42770178e-04]\n",
      " [ 9.75803472e-05  4.41611272e-04  9.83384352e-05 ... -1.00942522e-04\n",
      "   3.43533053e-04  3.42984682e-04]\n",
      " [-9.73913202e-05 -4.40868828e-04 -9.81005363e-05 ... -9.85217820e-05\n",
      "  -3.42936454e-04 -3.42337390e-04]]\n",
      "Training accuracy: 0.8991596638655462\n",
      "Test accuracy: 0.875\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "                   0.00      0.00      0.00         4\n",
      "          -1       0.88      1.00      0.93        35\n",
      "           1       0.00      0.00      0.00         1\n",
      "\n",
      "   micro avg       0.88      0.88      0.88        40\n",
      "   macro avg       0.29      0.33      0.31        40\n",
      "weighted avg       0.77      0.88      0.82        40\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\linear_model\\logistic.py:460: FutureWarning: Default multi_class will be changed to 'auto' in 0.22. Specify the multi_class option to silence this warning.\n",
      "  \"this warning.\", FutureWarning)\n",
      "C:\\Users\\iKhan\\Anaconda3\\envs\\nlp_course\\lib\\site-packages\\sklearn\\metrics\\classification.py:1143: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.\n",
      "  'precision', 'predicted', average, warn_for)\n"
     ]
    }
   ],
   "source": [
    "clf = LogisticRegression(penalty='l2', C=.0001, solver='liblinear')\n",
    "clf.fit(x_train[selected_feat], y_train)\n",
    "print('C:', c)\n",
    "print('Coefficient of each feature:', clf.coef_)\n",
    "print('Training accuracy:', clf.score(x_train[selected_feat], y_train))\n",
    "print('Test accuracy:', clf.score(x_test[selected_feat], y_test))\n",
    "print(classification_report(y_test, clf.predict(x_test[selected_feat])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('abil', 'inhibit')\n",
      "('absolut', 'configur')\n",
      "('acid', 'residu')\n",
      "('activ', 'among')\n",
      "('activ', 'analogu')\n",
      "('activ', 'compar')\n",
      "('activ', 'compound')\n",
      "('activ', 'demonstr')\n",
      "('activ', 'ic')\n",
      "('activ', 'mda')\n",
      "('activ', 'mycobacterium')\n",
      "('activ', 'panel')\n",
      "('activ', 'possibl')\n",
      "('activ', 'relationship')\n",
      "('activ', 'screen')\n",
      "('activ', 'sever')\n",
      "('activ', 'site')\n",
      "('activ', 'test')\n",
      "('activ', 'toward')\n",
      "('activ', 'use')\n",
      "('activ', 'well')\n",
      "('addit', 'display')\n",
      "('adp', 'ribos')\n",
      "('age', 'relat')\n",
      "('agonist', 'activ')\n",
      "('alloster', 'modul')\n",
      "('allow', 'identif')\n",
      "('almost', 'cell')\n",
      "('alzheim', 'diseas')\n",
      "('amid', 'deriv')\n",
      "('amino', 'acid')\n",
      "('among', 'compound')\n",
      "('among', 'test')\n",
      "('amyloid', 'aggreg')\n",
      "('analogu', 'evalu')\n",
      "('analogu', 'exhibit')\n",
      "('analogu', 'prepar')\n",
      "('analogu', 'synthes')\n",
      "('analysi', 'nmr')\n",
      "('anti', 'cancer')\n",
      "('anti', 'inflammatori')\n",
      "('anti', 'prolif')\n",
      "('antibacteri', 'activ')\n",
      "('antibacteri', 'antifung')\n",
      "('anticanc', 'activ')\n",
      "('anticanc', 'agent')\n",
      "('anticonvuls', 'activ')\n",
      "('antifung', 'activ')\n",
      "('antimicrobi', 'activ')\n",
      "('antioxid', 'activ')\n",
      "('antitumor', 'agent')\n",
      "('apoptot', 'cell')\n",
      "('appreci', 'cytotox')\n",
      "('aqueous', 'solubl')\n",
      "('arrest', 'cell')\n",
      "('bacteri', 'strain')\n",
      "('bactericid', 'activ')\n",
      "('base', 'drug')\n",
      "('best', 'compound')\n",
      "('beta', 'amyloid')\n",
      "('better', 'inhibitori')\n",
      "('betulin', 'acid')\n",
      "('bind', 'affin')\n",
      "('bind', 'site')\n",
      "('biochem', 'assay')\n",
      "('biolog', 'activ')\n",
      "('biolog', 'evalu')\n",
      "('bis', 'indol')\n",
      "('breast', 'cancer')\n",
      "('broad', 'spectrum')\n",
      "('cancer', 'cell')\n",
      "('cancer', 'drug')\n",
      "('cancer', 'properti')\n",
      "('cancer', 'therapi')\n",
      "('cancer', 'treatment')\n",
      "('carcinoma', 'cell')\n",
      "('cell', 'base')\n",
      "('cell', 'cycl')\n",
      "('cell', 'death')\n",
      "('cell', 'led')\n",
      "('cell', 'line')\n",
      "('cell', 'panel')\n",
      "('cell', 'signific')\n",
      "('cell', 'use')\n",
      "('cell', 'vitro')\n",
      "('cellular', 'activ')\n",
      "('chemic', 'convers')\n",
      "('chemic', 'investig')\n",
      "('chemic', 'librari')\n",
      "('chemic', 'probe')\n",
      "('chemic', 'structur')\n",
      "('circular', 'dichroism')\n",
      "('class', 'compound')\n",
      "('clerodan', 'diterpenoid')\n",
      "('clinic', 'candid')\n",
      "('close', 'relat')\n",
      "('co', 'receptor')\n",
      "('coloni', 'format')\n",
      "('compar', 'control')\n",
      "('compar', 'natur')\n",
      "('compar', 'posit')\n",
      "('compound', 'activ')\n",
      "('compound', 'confirm')\n",
      "('compound', 'contain')\n",
      "('compound', 'develop')\n",
      "('compound', 'display')\n",
      "('compound', 'effect')\n",
      "('compound', 'enhanc')\n",
      "('compound', 'evalu')\n",
      "('compound', 'exhibit')\n",
      "('compound', 'found')\n",
      "('compound', 'ic')\n",
      "('compound', 'induc')\n",
      "('compound', 'inhibit')\n",
      "('compound', 'led')\n",
      "('compound', 'may')\n",
      "('compound', 'potent')\n",
      "('compound', 'protect')\n",
      "('compound', 'purifi')\n",
      "('compound', 'screen')\n",
      "('compound', 'select')\n",
      "('compound', 'seri')\n",
      "('compound', 'show')\n",
      "('compound', 'shown')\n",
      "('compound', 'structur')\n",
      "('compound', 'test')\n",
      "('compound', 'within')\n",
      "('concentr', 'compound')\n",
      "('configur', 'compound')\n",
      "('configur', 'determin')\n",
      "('configur', 'establish')\n",
      "('coumarin', 'deriv')\n",
      "('coupl', 'differ')\n",
      "('critic', 'role')\n",
      "('crystal', 'ray')\n",
      "('cycl', 'phase')\n",
      "('cytotox', 'activ')\n",
      "('cytotox', 'effect')\n",
      "('data', 'absolut')\n",
      "('data', 'suggest')\n",
      "('deacetylas', 'hdac')\n",
      "('dehydrogenas', 'type')\n",
      "('demonstr', 'potent')\n",
      "('depend', 'manner')\n",
      "('deriv', 'bis')\n",
      "('deriv', 'design')\n",
      "('deriv', 'featur')\n",
      "('deriv', 'fungus')\n",
      "('deriv', 'less')\n",
      "('deriv', 'potent')\n",
      "('deriv', 'show')\n",
      "('describ', 'synthesi')\n",
      "('design', 'synthes')\n",
      "('design', 'synthesi')\n",
      "('determin', 'use')\n",
      "('develop', 'drug')\n",
      "('develop', 'new')\n",
      "('develop', 'potent')\n",
      "('develop', 'resist')\n",
      "('dien', 'yl')\n",
      "('diffract', 'analysi')\n",
      "('diffract', 'data')\n",
      "('diseas', 'ad')\n",
      "('diseas', 'develop')\n",
      "('diseas', 'modifi')\n",
      "('display', 'favor')\n",
      "('display', 'promis')\n",
      "('dock', 'simul')\n",
      "('dock', 'studi')\n",
      "('dose', 'depend')\n",
      "('drug', 'design')\n",
      "('drug', 'like')\n",
      "('drug', 'resist')\n",
      "('ec', 'valu')\n",
      "('ecd', 'calcul')\n",
      "('effici', 'one')\n",
      "('electron', 'circular')\n",
      "('element', 'analysi')\n",
      "('elucid', 'analysi')\n",
      "('emphas', 'import')\n",
      "('encourag', 'result')\n",
      "('enzym', 'famili')\n",
      "('enzym', 'inhibit')\n",
      "('epiderm', 'growth')\n",
      "('establish', 'use')\n",
      "('evalu', 'activ')\n",
      "('evalu', 'cytotox')\n",
      "('evalu', 'use')\n",
      "('exhibit', 'good')\n",
      "('exhibit', 'moder')\n",
      "('exhibit', 'potent')\n",
      "('exhibit', 'signific')\n",
      "('express', 'level')\n",
      "('factor', 'receptor')\n",
      "('favor', 'oral')\n",
      "('find', 'suggest')\n",
      "('first', 'exampl')\n",
      "('first', 'time')\n",
      "('five', 'known')\n",
      "('five', 'new')\n",
      "('focus', 'librari')\n",
      "('format', 'assay')\n",
      "('four', 'known')\n",
      "('function', 'contain')\n",
      "('function', 'includ')\n",
      "('function', 'may')\n",
      "('futur', 'studi')\n",
      "('gefitinib', 'among')\n",
      "('general', 'hydroxam')\n",
      "('gi', 'valu')\n",
      "('good', 'activ')\n",
      "('gram', 'posit')\n",
      "('group', 'higher')\n",
      "('group', 'show')\n",
      "('growth', 'factor')\n",
      "('growth', 'inhibit')\n",
      "('hdac', 'inhibitor')\n",
      "('hela', 'cell')\n",
      "('hepg', 'cell')\n",
      "('herein', 'describ')\n",
      "('herein', 'report')\n",
      "('heterocycl', 'scaffold')\n",
      "('high', 'affin')\n",
      "('high', 'potent')\n",
      "('high', 'select')\n",
      "('high', 'throughput')\n",
      "('histon', 'deacetylas')\n",
      "('hsd', 'inhibitor')\n",
      "('human', 'cancer')\n",
      "('human', 'ovarian')\n",
      "('hydrogen', 'bond')\n",
      "('hydroxam', 'acid')\n",
      "('hydroxysteroid', 'dehydrogenas')\n",
      "('ic', 'nm')\n",
      "('ic', 'valu')\n",
      "('identif', 'compound')\n",
      "('identifi', 'new')\n",
      "('identifi', 'potent')\n",
      "('identifi', 'select')\n",
      "('identifi', 'sever')\n",
      "('imidazo', 'pyridin')\n",
      "('import', 'function')\n",
      "('import', 'role')\n",
      "('improv', 'antibacteri')\n",
      "('improv', 'potenc')\n",
      "('includ', 'absolut')\n",
      "('includ', 'nmr')\n",
      "('induc', 'cell')\n",
      "('induc', 'seizur')\n",
      "('inflammatori', 'activ')\n",
      "('inhibit', 'tubulin')\n",
      "('inhibitor', 'result')\n",
      "('inhibitor', 'synthes')\n",
      "('inhibitori', 'activ')\n",
      "('inhibitori', 'effect')\n",
      "('inhibitori', 'potenc')\n",
      "('inhibitori', 'rate')\n",
      "('initi', 'hit')\n",
      "('interest', 'compound')\n",
      "('interpret', 'nmr')\n",
      "('ir', 'nmr')\n",
      "('isol', 'addit')\n",
      "('isol', 'cultur')\n",
      "('key', 'role')\n",
      "('kinas', 'domain')\n",
      "('known', 'compound')\n",
      "('lead', 'compound')\n",
      "('lead', 'optim')\n",
      "('led', 'discoveri')\n",
      "('led', 'identif')\n",
      "('led', 'isol')\n",
      "('less', 'activ')\n",
      "('less', 'cytotox')\n",
      "('like', 'properti')\n",
      "('line', 'analogu')\n",
      "('line', 'human')\n",
      "('line', 'result')\n",
      "('low', 'nanomolar')\n",
      "('lower', 'cytotox')\n",
      "('lu', 'human')\n",
      "('marin', 'deriv')\n",
      "('may', 'play')\n",
      "('may', 'potenti')\n",
      "('may', 'use')\n",
      "('mda', 'mb')\n",
      "('mechan', 'action')\n",
      "('medicin', 'chemistri')\n",
      "('methicillin', 'resist')\n",
      "('methyl', 'group')\n",
      "('mg', 'kg')\n",
      "('mic', 'valu')\n",
      "('ml', 'compar')\n",
      "('mode', 'action')\n",
      "('mode', 'bind')\n",
      "('model', 'reveal')\n",
      "('model', 'studi')\n",
      "('molecul', 'inhibitor')\n",
      "('molecular', 'dock')\n",
      "('molecular', 'model')\n",
      "('mous', 'model')\n",
      "('ms', 'data')\n",
      "('multidrug', 'resist')\n",
      "('mushroom', 'tyrosinas')\n",
      "('mycobacterium', 'tuberculosi')\n",
      "('natur', 'product')\n",
      "('new', 'class')\n",
      "('new', 'clerodan')\n",
      "('new', 'compound')\n",
      "('new', 'potenti')\n",
      "('newli', 'discov')\n",
      "('nm', 'almost')\n",
      "('nmr', 'experi')\n",
      "('nmr', 'ms')\n",
      "('nmr', 'spectroscop')\n",
      "('normal', 'cell')\n",
      "('novel', 'anti')\n",
      "('novel', 'class')\n",
      "('novel', 'function')\n",
      "('novel', 'mechan')\n",
      "('novel', 'seri')\n",
      "('nuclear', 'local')\n",
      "('obtain', 'two')\n",
      "('one', 'pot')\n",
      "('oral', 'pharmacokinet')\n",
      "('overcom', 'drug')\n",
      "('panel', 'human')\n",
      "('particular', 'compound')\n",
      "('pc', 'cell')\n",
      "('pc', 'mcf')\n",
      "('percentag', 'apoptot')\n",
      "('pharmacokinet', 'profil')\n",
      "('pharmacokinet', 'properti')\n",
      "('phenotyp', 'screen')\n",
      "('play', 'critic')\n",
      "('play', 'import')\n",
      "('play', 'key')\n",
      "('play', 'role')\n",
      "('poli', 'adp')\n",
      "('polymeras', 'parp')\n",
      "('potent', 'activ')\n",
      "('potent', 'anti')\n",
      "('potent', 'compound')\n",
      "('potent', 'cytotox')\n",
      "('potent', 'inhibit')\n",
      "('potent', 'inhibitor')\n",
      "('potent', 'select')\n",
      "('potent', 'tyrosinas')\n",
      "('potenti', 'drug')\n",
      "('potenti', 'lead')\n",
      "('potenti', 'target')\n",
      "('prepar', 'evalu')\n",
      "('present', 'studi')\n",
      "('previous', 'report')\n",
      "('probe', 'futur')\n",
      "('produc', 'new')\n",
      "('profil', 'compound')\n",
      "('promis', 'candid')\n",
      "('promis', 'lead')\n",
      "('promis', 'therapeut')\n",
      "('proteas', 'inhibitor')\n",
      "('protein', 'kinas')\n",
      "('provid', 'insight')\n",
      "('quinolin', 'deriv')\n",
      "('rang', 'ml')\n",
      "('rang', 'select')\n",
      "('ray', 'crystallographi')\n",
      "('ray', 'diffract')\n",
      "('receptor', 'ar')\n",
      "('receptor', 'egfr')\n",
      "('receptor', 'ht')\n",
      "('refer', 'compound')\n",
      "('refer', 'drug')\n",
      "('relationship', 'sar')\n",
      "('relationship', 'studi')\n",
      "('replic', 'ec')\n",
      "('report', 'design')\n",
      "('repres', 'compound')\n",
      "('repres', 'promis')\n",
      "('resist', 'tuberculosi')\n",
      "('result', 'identif')\n",
      "('result', 'indic')\n",
      "('result', 'suggest')\n",
      "('resveratrol', 'synthes')\n",
      "('ribos', 'polymeras')\n",
      "('ring', 'system')\n",
      "('sar', 'analysi')\n",
      "('sar', 'studi')\n",
      "('screen', 'campaign')\n",
      "('screen', 'compound')\n",
      "('screen', 'vitro')\n",
      "('secondari', 'metabolit')\n",
      "('seizur', 'model')\n",
      "('select', 'excel')\n",
      "('select', 'factor')\n",
      "('select', 'inhibit')\n",
      "('select', 'inhibitor')\n",
      "('seri', 'analog')\n",
      "('seri', 'identifi')\n",
      "('seri', 'novel')\n",
      "('serv', 'templat')\n",
      "('sever', 'compound')\n",
      "('sever', 'inhibitor')\n",
      "('show', 'activ')\n",
      "('show', 'inhibitori')\n",
      "('show', 'weak')\n",
      "('signific', 'cytotox')\n",
      "('signific', 'improv')\n",
      "('signific', 'inhibit')\n",
      "('signific', 'reduc')\n",
      "('simul', 'indic')\n",
      "('singl', 'crystal')\n",
      "('site', 'tubulin')\n",
      "('six', 'compound')\n",
      "('small', 'molecul')\n",
      "('spectroscop', 'data')\n",
      "('spectroscop', 'method')\n",
      "('staphylococcus', 'aureus')\n",
      "('start', 'point')\n",
      "('structur', 'activ')\n",
      "('structur', 'analogu')\n",
      "('structur', 'base')\n",
      "('structur', 'character')\n",
      "('structur', 'determin')\n",
      "('structur', 'elucid')\n",
      "('structur', 'featur')\n",
      "('structur', 'includ')\n",
      "('studi', 'compound')\n",
      "('studi', 'demonstr')\n",
      "('studi', 'gi')\n",
      "('studi', 'indic')\n",
      "('studi', 'present')\n",
      "('studi', 'provid')\n",
      "('studi', 'report')\n",
      "('studi', 'reveal')\n",
      "('studi', 'show')\n",
      "('studi', 'suggest')\n",
      "('studi', 'synthes')\n",
      "('studi', 'use')\n",
      "('substitut', 'aryl')\n",
      "('substitut', 'pattern')\n",
      "('success', 'identifi')\n",
      "('suggest', 'compound')\n",
      "('suitabl', 'vivo')\n",
      "('synthes', 'assay')\n",
      "('synthes', 'compound')\n",
      "('synthes', 'evalu')\n",
      "('synthes', 'screen')\n",
      "('synthes', 'via')\n",
      "('target', 'compound')\n",
      "('target', 'develop')\n",
      "('target', 'protein')\n",
      "('target', 'therapi')\n",
      "('tddft', 'ecd')\n",
      "('test', 'compound')\n",
      "('therapeut', 'agent')\n",
      "('therapeut', 'target')\n",
      "('three', 'cancer')\n",
      "('three', 'new')\n",
      "('three', 'step')\n",
      "('throughput', 'screen')\n",
      "('togeth', 'five')\n",
      "('tool', 'compound')\n",
      "('transcript', 'factor')\n",
      "('treatment', 'diseas')\n",
      "('tuberculosi', 'mtb')\n",
      "('tubulin', 'polymer')\n",
      "('tumor', 'cell')\n",
      "('tumor', 'growth')\n",
      "('two', 'new')\n",
      "('type', 'hsd')\n",
      "('type', 'wt')\n",
      "('tyrosinas', 'activ')\n",
      "('tyrosinas', 'inhibitor')\n",
      "('use', 'spectroscop')\n",
      "('use', 'treatment')\n",
      "('valu', 'nm')\n",
      "('valu', 'rang')\n",
      "('valu', 'respect')\n",
      "('valu', 'show')\n",
      "('various', 'human')\n",
      "('virus', 'replic')\n",
      "('vitro', 'activ')\n",
      "('vivo', 'efficaci')\n",
      "('water', 'solubl')\n",
      "('well', 'establish')\n",
      "('well', 'known')\n",
      "('well', 'toler')\n",
      "('wild', 'type')\n",
      "('xenograft', 'model')\n",
      "('yl', 'acet')\n",
      "abil\n",
      "abl\n",
      "absolut\n",
      "access\n",
      "account\n",
      "accumul\n",
      "acet\n",
      "acetoxi\n",
      "acetyl\n",
      "ach\n",
      "achiev\n",
      "acid\n",
      "act\n",
      "action\n",
      "activ\n",
      "ad\n",
      "addit\n",
      "address\n",
      "adenocarcinoma\n",
      "administ\n",
      "adp\n",
      "advantag\n",
      "affect\n",
      "affin\n",
      "afford\n",
      "age\n",
      "agent\n",
      "aggreg\n",
      "agonist\n",
      "aim\n",
      "akr\n",
      "aldehyd\n",
      "aliphat\n",
      "alkaloid\n",
      "alkyl\n",
      "alloster\n",
      "allow\n",
      "allyl\n",
      "almost\n",
      "along\n",
      "alpha\n",
      "alter\n",
      "although\n",
      "alzheim\n",
      "amid\n",
      "amin\n",
      "amino\n",
      "among\n",
      "amphotericin\n",
      "amyloid\n",
      "analog\n",
      "analogu\n",
      "analys\n",
      "analysi\n",
      "anim\n",
      "antagonist\n",
      "anti\n",
      "antibacteri\n",
      "antibiot\n",
      "anticanc\n",
      "anticonvuls\n",
      "antifung\n",
      "antimicrobi\n",
      "antioxid\n",
      "antitumor\n",
      "antivir\n",
      "apoptosi\n",
      "apoptot\n",
      "appli\n",
      "applic\n",
      "appreci\n",
      "approach\n",
      "approxim\n",
      "aqueous\n",
      "ar\n",
      "area\n",
      "aromat\n",
      "arrest\n",
      "aryl\n",
      "aspergillus\n",
      "assay\n",
      "assess\n",
      "assign\n",
      "associ\n",
      "atom\n",
      "attenu\n",
      "attract\n",
      "attribut\n",
      "aureus\n",
      "autoregulatori\n",
      "bacteri\n",
      "bacteria\n",
      "bactericid\n",
      "bacterium\n",
      "base\n",
      "basi\n",
      "bc\n",
      "bear\n",
      "benefici\n",
      "benefit\n",
      "benzo\n",
      "benzothiazol\n",
      "benzyl\n",
      "best\n",
      "beta\n",
      "better\n",
      "betulin\n",
      "biaryl\n",
      "bind\n",
      "bioassay\n",
      "bioavail\n",
      "biochem\n",
      "biogenet\n",
      "biolog\n",
      "biomark\n",
      "biosynthesi\n",
      "bis\n",
      "block\n",
      "blood\n",
      "bodi\n",
      "bond\n",
      "brain\n",
      "breast\n",
      "broad\n",
      "ca\n",
      "calcul\n",
      "campaign\n",
      "cancer\n",
      "candid\n",
      "carbon\n",
      "carbonyl\n",
      "carcinoma\n",
      "carri\n",
      "casearia\n",
      "caspas\n",
      "catalyt\n",
      "catalyz\n",
      "caus\n",
      "cc\n",
      "cdna\n",
      "cell\n",
      "cellular\n",
      "central\n",
      "certain\n",
      "chain\n",
      "chalcon\n",
      "challeng\n",
      "channel\n",
      "character\n",
      "characterist\n",
      "chelat\n",
      "chemic\n",
      "chemistri\n",
      "chlorid\n",
      "circular\n",
      "class\n",
      "classifi\n",
      "clerodan\n",
      "clinic\n",
      "clone\n",
      "close\n",
      "cluster\n",
      "cns\n",
      "co\n",
      "cognit\n",
      "collect\n",
      "coloni\n",
      "combin\n",
      "common\n",
      "compar\n",
      "complex\n",
      "compon\n",
      "compound\n",
      "comput\n",
      "concentr\n",
      "concern\n",
      "conclud\n",
      "condit\n",
      "conduct\n",
      "configur\n",
      "confirm\n",
      "conjug\n",
      "conserv\n",
      "consist\n",
      "contain\n",
      "content\n",
      "context\n",
      "continu\n",
      "contribut\n",
      "control\n",
      "convers\n",
      "copper\n",
      "core\n",
      "correl\n",
      "could\n",
      "coumarin\n",
      "coupl\n",
      "critic\n",
      "crystal\n",
      "crystallographi\n",
      "cultur\n",
      "current\n",
      "cycl\n",
      "cyp\n",
      "cystein\n",
      "cytotox\n",
      "data\n",
      "deacetylas\n",
      "death\n",
      "decreas\n",
      "defect\n",
      "dehydrogenas\n",
      "demonstr\n",
      "depend\n",
      "deriv\n",
      "describ\n",
      "design\n",
      "desir\n",
      "detail\n",
      "determin\n",
      "develop\n",
      "dgat\n",
      "dichroism\n",
      "dien\n",
      "differ\n",
      "differenti\n",
      "diffract\n",
      "dimer\n",
      "direct\n",
      "discov\n",
      "discoveri\n",
      "discuss\n",
      "diseas\n",
      "disord\n",
      "display\n",
      "disrupt\n",
      "dissoci\n",
      "distinct\n",
      "disubstitut\n",
      "diterpenoid\n",
      "divers\n",
      "dmso\n",
      "dna\n",
      "dock\n",
      "domain\n",
      "dorsal\n",
      "dose\n",
      "dpph\n",
      "dramat\n",
      "drosophila\n",
      "drug\n",
      "dual\n",
      "due\n",
      "earli\n",
      "ec\n",
      "ecd\n",
      "effect\n",
      "efficaci\n",
      "effici\n",
      "effort\n",
      "egfr\n",
      "eight\n",
      "either\n",
      "electron\n",
      "element\n",
      "elev\n",
      "elucid\n",
      "emphas\n",
      "employ\n",
      "enabl\n",
      "encod\n",
      "encourag\n",
      "endow\n",
      "enhanc\n",
      "enzym\n",
      "enzymat\n",
      "ep\n",
      "epiderm\n",
      "erk\n",
      "especi\n",
      "establish\n",
      "ester\n",
      "ether\n",
      "ethyl\n",
      "ev\n",
      "evalu\n",
      "eventu\n",
      "examin\n",
      "exampl\n",
      "excel\n",
      "except\n",
      "exert\n",
      "exhibit\n",
      "exist\n",
      "experi\n",
      "experiment\n",
      "explor\n",
      "exposur\n",
      "express\n",
      "extens\n",
      "extract\n",
      "factor\n",
      "famili\n",
      "favor\n",
      "featur\n",
      "final\n",
      "find\n",
      "first\n",
      "five\n",
      "flavonoid\n",
      "fluoresc\n",
      "fluorin\n",
      "fluoro\n",
      "focus\n",
      "fold\n",
      "follow\n",
      "form\n",
      "format\n",
      "found\n",
      "four\n",
      "fragment\n",
      "free\n",
      "fruit\n",
      "function\n",
      "fungal\n",
      "fungus\n",
      "furthermor\n",
      "futur\n",
      "gefitinib\n",
      "gene\n",
      "general\n",
      "generat\n",
      "genotox\n",
      "gi\n",
      "glycosid\n",
      "good\n",
      "gp\n",
      "gram\n",
      "greater\n",
      "group\n",
      "growth\n",
      "gsh\n",
      "guid\n",
      "hdac\n",
      "hela\n",
      "henc\n",
      "hepat\n",
      "hepatocellular\n",
      "hepg\n",
      "herein\n",
      "heterocycl\n",
      "high\n",
      "higher\n",
      "histon\n",
      "hit\n",
      "hiv\n",
      "homolog\n",
      "howev\n",
      "hsd\n",
      "ht\n",
      "hts\n",
      "human\n",
      "hydrazin\n",
      "hydrazon\n",
      "hydrogen\n",
      "hydrophob\n",
      "hydroxam\n",
      "hydroxi\n",
      "hydroxyl\n",
      "hydroxysteroid\n",
      "ic\n",
      "ident\n",
      "identif\n",
      "identifi\n",
      "ii\n",
      "ikk\n",
      "imidazo\n",
      "import\n",
      "improv\n",
      "inact\n",
      "includ\n",
      "inclus\n",
      "incorpor\n",
      "increas\n",
      "index\n",
      "indic\n",
      "indol\n",
      "induc\n",
      "induct\n",
      "infect\n",
      "inflammatori\n",
      "influenc\n",
      "inform\n",
      "inhibit\n",
      "inhibitor\n",
      "inhibitori\n",
      "initi\n",
      "inject\n",
      "insight\n",
      "interact\n",
      "interest\n",
      "intermedi\n",
      "interpret\n",
      "intracellular\n",
      "intraperiton\n",
      "introduc\n",
      "invas\n",
      "investig\n",
      "involv\n",
      "ir\n",
      "isoform\n",
      "isol\n",
      "kanamycin\n",
      "kb\n",
      "kda\n",
      "key\n",
      "kg\n",
      "ki\n",
      "kinas\n",
      "kinet\n",
      "knowledg\n",
      "known\n",
      "lack\n",
      "lactam\n",
      "larg\n",
      "ld\n",
      "lead\n",
      "least\n",
      "leav\n",
      "led\n",
      "length\n",
      "less\n",
      "leukemia\n",
      "level\n",
      "librari\n",
      "ligand\n",
      "like\n",
      "line\n",
      "linear\n",
      "link\n",
      "linkag\n",
      "linker\n",
      "lipid\n",
      "live\n",
      "liver\n",
      "local\n",
      "loop\n",
      "low\n",
      "lower\n",
      "lu\n",
      "luciferas\n",
      "lung\n",
      "macrocycl\n",
      "macrolid\n",
      "maintain\n",
      "major\n",
      "mani\n",
      "manner\n",
      "map\n",
      "marin\n",
      "mass\n",
      "matrix\n",
      "maxim\n",
      "may\n",
      "mb\n",
      "mcf\n",
      "mda\n",
      "mdr\n",
      "measur\n",
      "mechan\n",
      "media\n",
      "mediat\n",
      "medicin\n",
      "medium\n",
      "member\n",
      "membran\n",
      "meoh\n",
      "metabol\n",
      "metabolit\n",
      "metastasi\n",
      "methicillin\n",
      "method\n",
      "methyl\n",
      "mg\n",
      "mic\n",
      "mice\n",
      "micromolar\n",
      "microsom\n",
      "microwav\n",
      "might\n",
      "migrat\n",
      "minim\n",
      "ml\n",
      "mm\n",
      "mode\n",
      "model\n",
      "moder\n",
      "modif\n",
      "modifi\n",
      "modul\n",
      "moieti\n",
      "molecul\n",
      "molecular\n",
      "monom\n",
      "moreov\n",
      "motiv\n",
      "mous\n",
      "mrna\n",
      "ms\n",
      "mtb\n",
      "multi\n",
      "multicompon\n",
      "multidrug\n",
      "multipl\n",
      "mushroom\n",
      "mutant\n",
      "mutat\n",
      "mycobacteri\n",
      "mycobacterium\n",
      "myeloid\n",
      "myoblast\n",
      "na\n",
      "nanomolar\n",
      "nativ\n",
      "natur\n",
      "negat\n",
      "neurit\n",
      "neurolog\n",
      "neuron\n",
      "neurotox\n",
      "new\n",
      "newli\n",
      "nf\n",
      "nih\n",
      "nine\n",
      "nitro\n",
      "nitrogen\n",
      "nm\n",
      "nmr\n",
      "non\n",
      "normal\n",
      "note\n",
      "novel\n",
      "nuclear\n",
      "nucleus\n",
      "number\n",
      "observ\n",
      "obtain\n",
      "occur\n",
      "offer\n",
      "oh\n",
      "one\n",
      "opportun\n",
      "optim\n",
      "oral\n",
      "order\n",
      "ortho\n",
      "ortholog\n",
      "ovarian\n",
      "overal\n",
      "overcom\n",
      "overexpress\n",
      "oxid\n",
      "oxidas\n",
      "oxygen\n",
      "pak\n",
      "pancreat\n",
      "panel\n",
      "paper\n",
      "parasit\n",
      "parp\n",
      "particular\n",
      "pathogen\n",
      "patholog\n",
      "pathway\n",
      "patient\n",
      "pattern\n",
      "pc\n",
      "penetr\n",
      "peptid\n",
      "percentag\n",
      "perform\n",
      "permeabl\n",
      "pharmacokinet\n",
      "pharmacolog\n",
      "phase\n",
      "phenotyp\n",
      "phenyl\n",
      "phosphat\n",
      "phosphatas\n",
      "phosphoryl\n",
      "phosphotyrosin\n",
      "physiolog\n",
      "phytochem\n",
      "pi\n",
      "pk\n",
      "plant\n",
      "plasma\n",
      "play\n",
      "pocket\n",
      "podophyllotoxin\n",
      "point\n",
      "polar\n",
      "poli\n",
      "polymer\n",
      "polymeras\n",
      "posit\n",
      "possess\n",
      "possibl\n",
      "pot\n",
      "potenc\n",
      "potent\n",
      "potenti\n",
      "preclin\n",
      "predict\n",
      "prefer\n",
      "preliminari\n",
      "prepar\n",
      "presenc\n",
      "present\n",
      "preval\n",
      "prevent\n",
      "previous\n",
      "primari\n",
      "pro\n",
      "probabl\n",
      "probe\n",
      "problem\n",
      "process\n",
      "prodrug\n",
      "produc\n",
      "product\n",
      "profil\n",
      "progress\n",
      "prolif\n",
      "prolifer\n",
      "promis\n",
      "promot\n",
      "propens\n",
      "properti\n",
      "propos\n",
      "prostat\n",
      "proteas\n",
      "protect\n",
      "protein\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "protocol\n",
      "prove\n",
      "provid\n",
      "ptp\n",
      "purifi\n",
      "pyridin\n",
      "quaternari\n",
      "quinazolin\n",
      "quinolin\n",
      "radic\n",
      "rang\n",
      "rapid\n",
      "rare\n",
      "rat\n",
      "rate\n",
      "ration\n",
      "ray\n",
      "reaction\n",
      "recent\n",
      "receptor\n",
      "recombin\n",
      "recruit\n",
      "reduc\n",
      "reduct\n",
      "reductas\n",
      "refer\n",
      "regard\n",
      "region\n",
      "regul\n",
      "regulatori\n",
      "relat\n",
      "relationship\n",
      "releas\n",
      "relev\n",
      "remain\n",
      "remark\n",
      "replac\n",
      "replic\n",
      "report\n",
      "repres\n",
      "research\n",
      "residu\n",
      "resist\n",
      "respect\n",
      "respons\n",
      "result\n",
      "resveratrol\n",
      "reveal\n",
      "revis\n",
      "ribos\n",
      "rich\n",
      "ring\n",
      "rna\n",
      "robust\n",
      "role\n",
      "rout\n",
      "safeti\n",
      "sar\n",
      "satur\n",
      "scaffold\n",
      "screen\n",
      "sea\n",
      "search\n",
      "secondari\n",
      "seizur\n",
      "select\n",
      "sensit\n",
      "sequenc\n",
      "sequenti\n",
      "seri\n",
      "serv\n",
      "set\n",
      "seven\n",
      "sever\n",
      "sgf\n",
      "sh\n",
      "show\n",
      "shown\n",
      "side\n",
      "sif\n",
      "signal\n",
      "signific\n",
      "silico\n",
      "similar\n",
      "simul\n",
      "sinc\n",
      "singl\n",
      "site\n",
      "six\n",
      "size\n",
      "skeleton\n",
      "slight\n",
      "small\n",
      "solid\n",
      "solubl\n",
      "sp\n",
      "spare\n",
      "speci\n",
      "specif\n",
      "spectra\n",
      "spectroscop\n",
      "spectrum\n",
      "stabil\n",
      "stabl\n",
      "staphylococcus\n",
      "start\n",
      "state\n",
      "stem\n",
      "step\n",
      "steric\n",
      "steroid\n",
      "stilben\n",
      "stimul\n",
      "strain\n",
      "strand\n",
      "strategi\n",
      "streptomyc\n",
      "strong\n",
      "structur\n",
      "studi\n",
      "sub\n",
      "subject\n",
      "submicromolar\n",
      "subsequ\n",
      "substanti\n",
      "substitut\n",
      "substrat\n",
      "success\n",
      "suggest\n",
      "suitabl\n",
      "superior\n",
      "support\n",
      "suppress\n",
      "surfac\n",
      "surrog\n",
      "synthas\n",
      "synthes\n",
      "synthesi\n",
      "synthet\n",
      "system\n",
      "systemat\n",
      "target\n",
      "tddft\n",
      "techniqu\n",
      "templat\n",
      "ten\n",
      "term\n",
      "termin\n",
      "test\n",
      "therapeut\n",
      "therapi\n",
      "therebi\n",
      "therefor\n",
      "thiazol\n",
      "three\n",
      "throughput\n",
      "thus\n",
      "time\n",
      "tissu\n",
      "togeth\n",
      "toler\n",
      "tool\n",
      "total\n",
      "toward\n",
      "toxic\n",
      "transcript\n",
      "transfer\n",
      "translat\n",
      "treat\n",
      "treatment\n",
      "triazol\n",
      "tricycl\n",
      "triglycerid\n",
      "triterpen\n",
      "tryptophan\n",
      "tuberculosi\n",
      "tubulin\n",
      "tumor\n",
      "turn\n",
      "two\n",
      "type\n",
      "tyr\n",
      "tyrosin\n",
      "tyrosinas\n",
      "uniqu\n",
      "unknown\n",
      "unlik\n",
      "unmet\n",
      "unsatur\n",
      "unusu\n",
      "upon\n",
      "upregul\n",
      "urea\n",
      "us\n",
      "use\n",
      "util\n",
      "valid\n",
      "valu\n",
      "vari\n",
      "various\n",
      "via\n",
      "virus\n",
      "vitro\n",
      "vivo\n",
      "viz\n",
      "water\n",
      "weak\n",
      "weight\n",
      "well\n",
      "wherea\n",
      "whole\n",
      "wide\n",
      "widespread\n",
      "wild\n",
      "within\n",
      "work\n",
      "wt\n",
      "xenograft\n",
      "xenopus\n",
      "year\n",
      "yeast\n",
      "yield\n",
      "yl\n"
     ]
    }
   ],
   "source": [
    "#f = open('names.txt', 'w')\n",
    "\n",
    "for col in x_train[selected_feat].columns: \n",
    "    print(col) \n",
    "  #  f.write(col + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
